{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "c04f74c2",
   "metadata": {},
   "source": [
    "## Building the ClinVar TP53 Dataset\n",
    "---\n",
    "\n",
    "#### **Project Objectives**:\n",
    "\n",
    "**1) Downloading the raw ClinVar dataset**\n",
    "\n",
    "- Downloaded from: [https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/archive/](https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/archive/)\n",
    "- Loading the dataset into a pandas DataFrame.\n",
    "- Reviewing columns and data types.\n",
    "- Identifying potential features.\n",
    "\n",
    "**2) Initial data cleaning (`1`)**\n",
    "\n",
    "- Filtering for only the **TP53** target gene.\n",
    "- Removing duplicates and incomplete variant records.\n",
    "- Removing metadata (non-informative features lacking biological relevance).\n",
    "- Identifying missing values (NaN) and applying imputation methods.\n",
    "\n",
    "**3) Engineering the target variable (binary classification)**\n",
    "\n",
    "- For binary classification, focusing solely on **benign** and **pathogenic** labels.\n",
    "- Removing all other variant types: **VUS**, **conflicting**, and **-1**.\n",
    "- Mapping: `likely benign` → **benign** and `likely pathogenic` → **pathogenic**.\n",
    "\n",
    "**4) Integrating VEP (Variant Effect Predictor) annotations**\n",
    "\n",
    "- Using the Ensembl REST API to fetch additional annotations: [https://rest.ensembl.org/#VEP](https://rest.ensembl.org/#VEP)\n",
    "- Goal: Create an extended dataset with enriched annotations to improve subsequent model performance.\n",
    "\n",
    "**5) Cleaning the extended dataset (`2`)**\n",
    "\n",
    "- Removing variants containing 'NA' values.\n",
    "- Filtering out variants without any VEP annotations.\n",
    "- Performing complete encoding of categorical annotations.\n",
    "\n",
    "**6) Forming the final feature set**\n",
    "\n",
    "- Defining a minimal, interpretable set of features.\n",
    "- Finalizing a cleaned dataset ready for training various ML models."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "74461780",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "a001b9eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "#libraries for this phase\n",
    "from pathlib import Path\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os\n",
    "from pathlib import Path"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dd4213b3",
   "metadata": {},
   "source": [
    "Local project folder structure:\n",
    "\n",
    "Project root: Binary Classification\n",
    "\n",
    "Folders: raw data and notebooks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a1c656ff",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "c:\\Users\\Admin\\Binary Classification\\raw data\n"
     ]
    }
   ],
   "source": [
    "current = os.getcwd()\n",
    "parent_dir = Path.cwd().parent\n",
    "for root,dirs,files in os.walk(parent_dir):\n",
    "    for file in files:\n",
    "        if file.endswith('_raw.txt'):\n",
    "            print(root)\n",
    "            raw_file_path = os.path.join(root,file)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f4b1578e",
   "metadata": {},
   "outputs": [],
   "source": [
    "raw_data_iterator = pd.read_csv(\n",
    "    raw_file_path,\n",
    "    sep='\\t',\n",
    "    chunksize=200000,\n",
    "    iterator=True\n",
    ")\n",
    "#TexTfile obj because of the file size"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "5bc4e62e",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Admin\\AppData\\Local\\Temp\\ipykernel_12468\\4095176858.py:2: DtypeWarning: Columns (18) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  for chunk in raw_data_iterator:\n",
      "C:\\Users\\Admin\\AppData\\Local\\Temp\\ipykernel_12468\\4095176858.py:2: DtypeWarning: Columns (18) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  for chunk in raw_data_iterator:\n",
      "C:\\Users\\Admin\\AppData\\Local\\Temp\\ipykernel_12468\\4095176858.py:2: DtypeWarning: Columns (18) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  for chunk in raw_data_iterator:\n",
      "C:\\Users\\Admin\\AppData\\Local\\Temp\\ipykernel_12468\\4095176858.py:2: DtypeWarning: Columns (18) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  for chunk in raw_data_iterator:\n",
      "C:\\Users\\Admin\\AppData\\Local\\Temp\\ipykernel_12468\\4095176858.py:2: DtypeWarning: Columns (18) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  for chunk in raw_data_iterator:\n",
      "C:\\Users\\Admin\\AppData\\Local\\Temp\\ipykernel_12468\\4095176858.py:2: DtypeWarning: Columns (18) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  for chunk in raw_data_iterator:\n",
      "C:\\Users\\Admin\\AppData\\Local\\Temp\\ipykernel_12468\\4095176858.py:2: DtypeWarning: Columns (18) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  for chunk in raw_data_iterator:\n",
      "C:\\Users\\Admin\\AppData\\Local\\Temp\\ipykernel_12468\\4095176858.py:2: DtypeWarning: Columns (18) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  for chunk in raw_data_iterator:\n",
      "C:\\Users\\Admin\\AppData\\Local\\Temp\\ipykernel_12468\\4095176858.py:2: DtypeWarning: Columns (18) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  for chunk in raw_data_iterator:\n",
      "C:\\Users\\Admin\\AppData\\Local\\Temp\\ipykernel_12468\\4095176858.py:2: DtypeWarning: Columns (18) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  for chunk in raw_data_iterator:\n"
     ]
    }
   ],
   "source": [
    "dataset = []\n",
    "for chunk in raw_data_iterator:\n",
    "    dataset.append(chunk)\n",
    "dataset = pd.concat(dataset)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "9b86aadd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(8364862, 43)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataset.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "c8cc0b1f",
   "metadata": {},
   "outputs": [],
   "source": [
    "dataset = dataset[dataset['GeneSymbol']=='TP53'] #tp53 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "4c5ee1f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter dataset to keep only GRCh38 assembly (newer version)\n",
    "# Rationale: GRCh38 contains updated genome annotations and corrections compared to GRCh37\n",
    "dataset = dataset[dataset['Assembly'] == 'GRCh38']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "cce57477",
   "metadata": {},
   "outputs": [],
   "source": [
    "dataset = dataset[dataset['Assembly']=='GRCh38']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "64ec22de",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Assembly\n",
       "GRCh38    3758\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataset['Assembly'].value_counts() # GRCh38 - 3758\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "d43e2844",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "np.False_"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataset.duplicated().any()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "d1bd9885",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "np.int64(0)"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataset.isnull().any()\n",
    "dataset.isna().any().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "d39df272",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['#AlleleID', 'Type', 'Name', 'GeneID', 'GeneSymbol', 'HGNC_ID',\n",
       "       'ClinicalSignificance', 'ClinSigSimple', 'LastEvaluated', 'RS# (dbSNP)',\n",
       "       'nsv/esv (dbVar)', 'RCVaccession', 'PhenotypeIDS', 'PhenotypeList',\n",
       "       'Origin', 'OriginSimple', 'Assembly', 'ChromosomeAccession',\n",
       "       'Chromosome', 'Start', 'Stop', 'ReferenceAllele', 'AlternateAllele',\n",
       "       'Cytogenetic', 'ReviewStatus', 'NumberSubmitters', 'Guidelines',\n",
       "       'TestedInGTR', 'OtherIDs', 'SubmitterCategories', 'VariationID',\n",
       "       'PositionVCF', 'ReferenceAlleleVCF', 'AlternateAlleleVCF',\n",
       "       'SomaticClinicalImpact', 'SomaticClinicalImpactLastEvaluated',\n",
       "       'ReviewStatusClinicalImpact', 'Oncogenicity',\n",
       "       'OncogenicityLastEvaluated', 'ReviewStatusOncogenicity',\n",
       "       'SCVsForAggregateGermlineClassification',\n",
       "       'SCVsForAggregateSomaticClinicalImpact',\n",
       "       'SCVsForAggregateOncogenicityClassification'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataset.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dc105ef4",
   "metadata": {},
   "source": [
    "##### 2) Initial Dataset Cleaning\n",
    "\n",
    "> Dropping all columns except the identification columns necessary for subsequent VEP annotation integration.\n",
    "\n",
    "- **Chromosome** - Required for precise genomic location\n",
    "- **Start** - Unique variant position (I'm only tracking single nucleotide changes, so the **Stop** column is not needed)\n",
    "- **ReferenceAlleleVCF** - Reference nucleotide\n",
    "- **AlternativeAlleleVCF** - Alternative/mutated nucleotide  \n",
    "- **GeneSymbol** - Confirmation that the variant is from the target gene in this project (**TP53**)\n",
    "- **ClinSigSimple** - Binary label (0 and 1)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7faccc03",
   "metadata": {},
   "source": [
    "[Chromosome, Start, ReferenceAlleleVCF, AlternateAlleleVCF]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "b4299311",
   "metadata": {},
   "outputs": [],
   "source": [
    "selected_cols = ['Chromosome','Start','ReferenceAlleleVCF','AlternateAlleleVCF','GeneSymbol','ClinSigSimple']\n",
    "dataset = dataset[selected_cols] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "90536198",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chromosome            False\n",
       "Start                 False\n",
       "ReferenceAlleleVCF    False\n",
       "AlternateAlleleVCF    False\n",
       "GeneSymbol            False\n",
       "ClinSigSimple         False\n",
       "dtype: bool"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataset.isna().any()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "36c44144",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chromosome            False\n",
       "Start                 False\n",
       "ReferenceAlleleVCF    False\n",
       "AlternateAlleleVCF    False\n",
       "GeneSymbol            False\n",
       "ClinSigSimple         False\n",
       "dtype: bool"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataset.isnull().any()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "7f9ea314",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chromosome\n",
       "17    3674\n",
       "17      84\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataset['Chromosome'].value_counts() "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "46037f47",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Standardizing the 'Chromosome' column, merging mixed dtype (str) and (int) into `str` type\n",
    "dataset['Chromosome'] = dataset['Chromosome'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "4cf6f396",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Chromosome\n",
       "17    3758\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataset['Chromosome'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "89ef3d75",
   "metadata": {},
   "outputs": [],
   "source": [
    "# odbacivanje svih vrednosti koji su 'na'\n",
    "dataset = dataset[dataset['Chromosome']=='17']\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2f88c305",
   "metadata": {},
   "source": [
    "### `ReferenceAlleleVCF` and `AlternateAlleleVCF` contain specific nucleotides (A/C/G/T) and are aligned with the reference genome. They are precise sources of information and essential for VEP annotation.\n",
    "\n",
    "### These two columns require additional cleaning as they contain not only single characters but also longer sequences. This indicates variant types like insertions and deletions affecting larger gene segments, which are not the current focus of this research. The focus is on single nucleotide variants (SNVs) and determining their pathogenicity.\n",
    "\n",
    "#### Alternative: Also filter for valid nucleotides only"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "77b36c2b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ReferenceAlleleVCF\n",
       "G                           948\n",
       "C                           890\n",
       "A                           576\n",
       "T                           555\n",
       "GC                           41\n",
       "                           ... \n",
       "GGAGTCTTCCAGTGTGATGATGGT      1\n",
       "ACTATGTCG                     1\n",
       "GCGGCTCATAGGGCACC             1\n",
       "AGTTGGCAAAACATCTT             1\n",
       "ATGTAGT                       1\n",
       "Name: count, Length: 416, dtype: int64"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataset['ReferenceAlleleVCF'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "b69a0266",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ReferenceAlleleVCF\n",
       "G    852\n",
       "C    812\n",
       "A    515\n",
       "T    503\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Keeping the rows with only 1 Character AKA nucleotide \n",
    "dataset = dataset[(dataset['ReferenceAlleleVCF'].str.len()==1)&(dataset['AlternateAlleleVCF'].str.len()==1)]\n",
    "dataset['ReferenceAlleleVCF'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "c9bcacd4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "AlternateAlleleVCF\n",
       "A    778\n",
       "T    704\n",
       "C    632\n",
       "G    568\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataset['AlternateAlleleVCF'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "eb0e3100",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GeneSymbol\n",
       "TP53    2682\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataset['GeneSymbol'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "b348ab4a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start</th>\n",
       "      <th>ReferenceAlleleVCF</th>\n",
       "      <th>AlternateAlleleVCF</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>ClinSigSimple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17</td>\n",
       "      <td>7674221</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>17</td>\n",
       "      <td>7674191</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>17</td>\n",
       "      <td>7674230</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>17</td>\n",
       "      <td>7674208</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>17</td>\n",
       "      <td>7676154</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2677</th>\n",
       "      <td>17</td>\n",
       "      <td>7676274</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>TP53</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2678</th>\n",
       "      <td>17</td>\n",
       "      <td>7676351</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>TP53</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2679</th>\n",
       "      <td>17</td>\n",
       "      <td>7676405</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>TP53</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2680</th>\n",
       "      <td>17</td>\n",
       "      <td>7676517</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>TP53</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2681</th>\n",
       "      <td>17</td>\n",
       "      <td>7673785</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>TP53</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2682 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chromosome    Start ReferenceAlleleVCF AlternateAlleleVCF GeneSymbol  \\\n",
       "0            17  7674221                  G                  A       TP53   \n",
       "1            17  7674191                  C                  T       TP53   \n",
       "2            17  7674230                  C                  A       TP53   \n",
       "3            17  7674208                  A                  G       TP53   \n",
       "4            17  7676154                  G                  C       TP53   \n",
       "...         ...      ...                ...                ...        ...   \n",
       "2677         17  7676274                  T                  A       TP53   \n",
       "2678         17  7676351                  T                  A       TP53   \n",
       "2679         17  7676405                  T                  G       TP53   \n",
       "2680         17  7676517                  T                  C       TP53   \n",
       "2681         17  7673785                  C                  A       TP53   \n",
       "\n",
       "      ClinSigSimple  \n",
       "0                 1  \n",
       "1                 1  \n",
       "2                 1  \n",
       "3                 1  \n",
       "4                 1  \n",
       "...             ...  \n",
       "2677             -1  \n",
       "2678             -1  \n",
       "2679             -1  \n",
       "2680             -1  \n",
       "2681             -1  \n",
       "\n",
       "[2682 rows x 6 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataset = dataset.reset_index(drop=True)\n",
    "dataset"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "570e3139",
   "metadata": {},
   "source": [
    "Further filter the dataset according to the coordinates/location of the gene (TP53) provided on the official Ensembl website [https://www.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000141510;r=17:7661779-7687546]\n",
    "\n",
    "**location** - Chromosome 17: 7,661,779-7,687,546 reverse strand."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "029c4009",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start</th>\n",
       "      <th>ReferenceAlleleVCF</th>\n",
       "      <th>AlternateAlleleVCF</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>ClinSigSimple</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17</td>\n",
       "      <td>7674221</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>17</td>\n",
       "      <td>7674191</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>17</td>\n",
       "      <td>7674230</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>17</td>\n",
       "      <td>7674208</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>17</td>\n",
       "      <td>7676154</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2677</th>\n",
       "      <td>17</td>\n",
       "      <td>7676274</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>TP53</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2678</th>\n",
       "      <td>17</td>\n",
       "      <td>7676351</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>TP53</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2679</th>\n",
       "      <td>17</td>\n",
       "      <td>7676405</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>TP53</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2680</th>\n",
       "      <td>17</td>\n",
       "      <td>7676517</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>TP53</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2681</th>\n",
       "      <td>17</td>\n",
       "      <td>7673785</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>TP53</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2678 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chromosome    Start ReferenceAlleleVCF AlternateAlleleVCF GeneSymbol  \\\n",
       "0            17  7674221                  G                  A       TP53   \n",
       "1            17  7674191                  C                  T       TP53   \n",
       "2            17  7674230                  C                  A       TP53   \n",
       "3            17  7674208                  A                  G       TP53   \n",
       "4            17  7676154                  G                  C       TP53   \n",
       "...         ...      ...                ...                ...        ...   \n",
       "2677         17  7676274                  T                  A       TP53   \n",
       "2678         17  7676351                  T                  A       TP53   \n",
       "2679         17  7676405                  T                  G       TP53   \n",
       "2680         17  7676517                  T                  C       TP53   \n",
       "2681         17  7673785                  C                  A       TP53   \n",
       "\n",
       "      ClinSigSimple  \n",
       "0                 1  \n",
       "1                 1  \n",
       "2                 1  \n",
       "3                 1  \n",
       "4                 1  \n",
       "...             ...  \n",
       "2677             -1  \n",
       "2678             -1  \n",
       "2679             -1  \n",
       "2680             -1  \n",
       "2681             -1  \n",
       "\n",
       "[2678 rows x 6 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tp53_start = 7661779\n",
    "tp53_end = 7687546\n",
    "\n",
    "dataset = dataset[\n",
    "    (dataset['Start']>=tp53_start) &\n",
    "    (dataset['Start']<=tp53_end)\n",
    "]\n",
    "\n",
    "dataset"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "92e19d3d",
   "metadata": {},
   "source": [
    "### 3) Target feature -> __ClinSigSimple__\n",
    "- label 0 `benign`/`likely bening` -> **benign**\n",
    "\n",
    "- label 1 `pathogenic`/`likely pathogenic` -> **pathogenic**\n",
    "\n",
    "- label -1 ‘‘conflicting interpretation of pathogenicity’’  variants without clear effects on the carrier (drop)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "6158d4b9",
   "metadata": {},
   "outputs": [],
   "source": [
    "dataset = dataset[dataset['ClinSigSimple']!=-1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "90d2302f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ClinSigSimple\n",
       "0    2026\n",
       "1     621\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataset['ClinSigSimple'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d0ca048d",
   "metadata": {},
   "source": [
    "#### 4) Integration of VEP variants (ENSEBML REST API) [ https://rest.ensembl.org/documentation/info/vep_region_post ]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c3770f9f",
   "metadata": {},
   "source": [
    "In practice, the Ensembl VEP returns multiple annotations per variant because a single variant can have multiple transcripts.\n",
    "\n",
    "In scientific papers, the practice is to use a representative annotation per variant, marked as canonical. This way, I would establish uniformity in the data I retrieve for the variants in the current dataset.\n",
    "\n",
    "I also came across this rule in the paper 'Annotating and prioritizing genomic variants using the Ensembl Variant Effect Predictor—A tutorial',\n",
    "\n",
    "Hunt SE et al. (2021) [https://pmc.ncbi.nlm.nih.gov/articles/PMC7613081/]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5d28f7c6",
   "metadata": {},
   "source": [
    "The `plugin` section on the page refers specifically to the additional annotations I will integrate into the dataset.\n",
    "\n",
    "- AlphaMissense\n",
    "\n",
    "-  Blosum62 \n",
    "\n",
    "- CADD (cadd_raw & cadd_phred)\n",
    "\n",
    "-  LoF \n",
    "\n",
    "- PolyPhen score\n",
    "\n",
    "- Sift_score\n",
    "\n",
    "+ IMPACT which request returns always.\n",
    "\n",
    "After several unsuccessful attempts with the GET method for a single variant, I tried a POST request [https://rest.ensembl.org/documentation/info/vep_region_post] to retrieve annotations for a larger number of variants.\n",
    "\n",
    "The format used for the POST request is: [Chromosome Start . ReferenceAlleleVCF AlternateAlleleVCF]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "71e5e3e5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2647\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Admin\\AppData\\Local\\Temp\\ipykernel_12468\\3547526549.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  dataset['vcf_format'] = dataset.apply(\n"
     ]
    }
   ],
   "source": [
    "\n",
    "dataset['vcf_format'] = dataset.apply(\n",
    "    lambda row: f\"{row['Chromosome']} {row['Start']} . {row['ReferenceAlleleVCF']} {row['AlternateAlleleVCF']}\", \n",
    "    axis=1\n",
    ")\n",
    "\n",
    "variantsParams = dataset['vcf_format'].to_list()\n",
    "print(len(variantsParams))\n",
    "_test = variantsParams\n",
    "# 50 test"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9e19320e",
   "metadata": {},
   "source": [
    "For the API call, I followed the official example at the bottom of the page [https://rest.ensembl.org/documentation/info/vep_region_post] with minor modifications.\n",
    "\n",
    "In the param variable, I defined the final set of available parameters after several attempts."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "0bbc68ee",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 7674221 . G A\n",
      "17 7674191 . C T\n",
      "17 7674230 . C A\n",
      "17 7674208 . A G\n",
      "17 7676154 . G C\n",
      "17 7674216 . C A\n",
      "17 7675143 . C A\n",
      "17 7674238 . C T\n",
      "17 7674229 . C T\n",
      "17 7674220 . C T\n",
      "17 7675214 . A G\n",
      "17 7673806 . C A\n",
      "17 7674241 . G A\n",
      "17 7676011 . T A\n",
      "17 7673776 . G A\n",
      "17 7674230 . C T\n",
      "17 7673802 . C T\n",
      "17 7673554 . C A\n",
      "17 7673781 . C G\n",
      "17 7675161 . G T\n",
      "17 7675161 . G A\n",
      "17 7676264 . C A\n",
      "17 7674193 . A T\n",
      "17 7675088 . C T\n",
      "17 7670678 . A G\n",
      "17 7675200 . C G\n",
      "17 7673745 . T A\n",
      "17 7670699 . C T\n",
      "17 7674965 . G A\n",
      "17 7674872 . T G\n",
      "17 7673766 . T A\n",
      "17 7668434 . T G\n",
      "17 7675070 . C A\n",
      "17 7676564 . C G\n",
      "17 7670669 . G T\n",
      "17 7676230 . G A\n",
      "17 7674945 . G A\n",
      "17 7674894 . G A\n",
      "17 7674892 . T C\n",
      "17 7673803 . G A\n",
      "17 7673610 . T C\n",
      "17 7670658 . T C\n",
      "17 7670616 . G A\n",
      "17 7674859 . C T\n",
      "17 7675212 . A G\n",
      "17 7673728 . C A\n",
      "17 7673523 . A G\n",
      "17 7674944 . C A\n",
      "17 7674227 . T C\n",
      "17 7673733 . T C\n",
      "17 7687373 . T G\n",
      "17 7676196 . G C\n",
      "17 7676121 . G A\n",
      "17 7676567 . C T\n",
      "17 7676040 . C T\n",
      "17 7675146 . G A\n",
      "17 7675145 . C T\n",
      "17 7675140 . G A\n",
      "17 7675127 . A C\n",
      "17 7675124 . T C\n",
      "17 7675077 . G A\n",
      "17 7674959 . G C\n",
      "17 7674951 . G A\n",
      "17 7674927 . G A\n",
      "17 7674872 . T C\n",
      "17 7674262 . T C\n",
      "17 7674259 . T C\n",
      "17 7674226 . A C\n",
      "17 7673820 . C T\n",
      "17 7673773 . G A\n",
      "17 7673751 . C T\n",
      "17 7673743 . C A\n",
      "17 7676387 . C T\n",
      "17 7676261 . C T\n",
      "17 7673735 . A G\n",
      "17 7674865 . C A\n",
      "17 7673734 . G A\n",
      "17 7673559 . C T\n",
      "17 7675226 . G A\n",
      "17 7675152 . C T\n",
      "17 7675150 . G A\n",
      "17 7674824 . C A\n",
      "17 7674828 . T C\n",
      "17 7674926 . C T\n",
      "17 7674893 . C T\n",
      "17 7670613 . A C\n",
      "17 7673312 . A C\n",
      "17 7676224 . C G\n",
      "17 7675104 . T C\n",
      "17 7674949 . A G\n",
      "17 7676483 . G C\n",
      "17 7674164 . G A\n",
      "17 7676574 . A T\n",
      "17 7676050 . A G\n",
      "17 7675190 . C T\n",
      "17 7673776 . G C\n",
      "17 7676119 . C T\n",
      "17 7676041 . G T\n",
      "17 7674889 . A C\n",
      "17 7674197 . T C\n",
      "17 7676225 . G T\n",
      "17 7676011 . T C\n",
      "17 7676005 . C T\n",
      "17 7673593 . G C\n",
      "17 7676055 . C T\n",
      "17 7673778 . T C\n",
      "17 7670706 . G A\n",
      "17 7674268 . A G\n",
      "17 7673836 . C T\n",
      "17 7676032 . A C\n",
      "17 7670627 . C T\n",
      "17 7669692 . T C\n",
      "17 7675058 . C T\n",
      "17 7673609 . C T\n",
      "17 7669628 . T G\n",
      "17 7676244 . T C\n",
      "17 7673728 . C T\n",
      "17 7676148 . G A\n",
      "17 7675101 . C T\n",
      "17 7669671 . C T\n",
      "17 7676262 . G T\n",
      "17 7675208 . C T\n",
      "17 7673821 . G A\n",
      "17 7673761 . C T\n",
      "17 7674218 . T A\n",
      "17 7670694 . C G\n",
      "17 7676121 . G T\n",
      "17 7675139 . C T\n",
      "17 7676113 . C T\n",
      "17 7675221 . T A\n",
      "17 7670630 . C G\n",
      "17 7674887 . C G\n",
      "17 7675178 . A G\n",
      "17 7670637 . C T\n",
      "17 7674193 . A C\n",
      "17 7673704 . G A\n",
      "17 7670716 . C T\n",
      "17 7674257 . A C\n",
      "17 7676041 . G A\n",
      "17 7674217 . C T\n",
      "17 7670711 . C T\n",
      "17 7675070 . C T\n",
      "17 7673772 . C T\n",
      "17 7673713 . T C\n",
      "17 7670596 . T C\n",
      "17 7676152 . C T\n",
      "17 7676049 . T G\n",
      "17 7676047 . C T\n",
      "17 7670700 . G A\n",
      "17 7670630 . C A\n",
      "17 7676264 . C G\n",
      "17 7675071 . G A\n",
      "17 7675185 . C T\n",
      "17 7676576 . C G\n",
      "17 7676049 . T C\n",
      "17 7673710 . T C\n",
      "17 7674252 . C T\n",
      "17 7675157 . G A\n",
      "17 7670699 . C A\n",
      "17 7670694 . C T\n",
      "17 7676155 . G C\n",
      "17 7676198 . G T\n",
      "17 7673833 . T C\n",
      "17 7676250 . A G\n",
      "17 7669626 . C A\n",
      "17 7674866 . G A\n",
      "17 7673781 . C A\n",
      "17 7673823 . C T\n",
      "17 7674249 . A C\n",
      "17 7675131 . C T\n",
      "17 7674241 . G C\n",
      "17 7676154 . G G\n",
      "17 7670699 . C G\n",
      "17 7675994 . C T\n",
      "17 7669641 . T C\n",
      "17 7670685 . G A\n",
      "17 7670695 . G A\n",
      "17 7670709 . C G\n",
      "17 7670732 . G A\n",
      "17 7673522 . C G\n",
      "17 7673775 . C A\n",
      "17 7673778 . T A\n",
      "17 7674169 . G A\n",
      "17 7674211 . A C\n",
      "17 7674212 . T G\n",
      "17 7674235 . A G\n",
      "17 7674250 . C T\n",
      "17 7674254 . T C\n",
      "17 7674256 . T G\n",
      "17 7674841 . C G\n",
      "17 7674885 . C T\n",
      "17 7674924 . C T\n",
      "17 7674926 . C A\n",
      "17 7674956 . T C\n",
      "17 7675088 . C A\n",
      "17 7675119 . G A\n",
      "17 7675137 . C G\n",
      "17 7675232 . G A\n",
      "17 7675988 . A G\n",
      "17 7675999 . A T\n",
      "17 7676025 . T C\n",
      "17 7676079 . A T\n",
      "17 7676080 . C G\n",
      "17 7676103 . G A\n",
      "17 7676124 . G A\n",
      "17 7676181 . G A\n",
      "17 7676263 . G A\n",
      "17 7676281 . G A\n",
      "17 7676577 . T G\n",
      "17 7674882 . C T\n",
      "17 7669642 . G A\n",
      "17 7669654 . G A\n",
      "17 7669656 . G A\n",
      "17 7669656 . G T\n",
      "17 7670631 . C T\n",
      "17 7670707 . C T\n",
      "17 7670709 . C T\n",
      "17 7670712 . G A\n",
      "17 7673585 . A T\n",
      "17 7673604 . C T\n",
      "17 7673613 . G A\n",
      "17 7673717 . T C\n",
      "17 7673729 . G A\n",
      "17 7673764 . C T\n",
      "17 7673768 . T G\n",
      "17 7673783 . C T\n",
      "17 7673790 . C T\n",
      "17 7673793 . G C\n",
      "17 7673806 . C T\n",
      "17 7674181 . C G\n",
      "17 7674195 . T C\n",
      "17 7674222 . G A\n",
      "17 7674223 . T A\n",
      "17 7674227 . T G\n",
      "17 7674236 . T G\n",
      "17 7674267 . G A\n",
      "17 7674865 . C T\n",
      "17 7674870 . C T\n",
      "17 7674874 . G T\n",
      "17 7674877 . C T\n",
      "17 7674913 . C T\n",
      "17 7674916 . A T\n",
      "17 7674919 . C T\n",
      "17 7674953 . T A\n",
      "17 7674953 . T C\n",
      "17 7675054 . A G\n",
      "17 7675057 . G A\n",
      "17 7675085 . C T\n",
      "17 7675102 . C T\n",
      "17 7675103 . G A\n",
      "17 7675125 . A C\n",
      "17 7675138 . G A\n",
      "17 7675143 . C T\n",
      "17 7675144 . G A\n",
      "17 7675147 . G T\n",
      "17 7675148 . G C\n",
      "17 7675176 . A C\n",
      "17 7675179 . G A\n",
      "17 7675180 . C T\n",
      "17 7675181 . T G\n",
      "17 7675185 . C A\n",
      "17 7675193 . G T\n",
      "17 7675199 . G A\n",
      "17 7675204 . T C\n",
      "17 7675204 . T G\n",
      "17 7675222 . G A\n",
      "17 7675238 . T C\n",
      "17 7675995 . G A\n",
      "17 7676015 . T A\n",
      "17 7676064 . G A\n",
      "17 7676102 . G A\n",
      "17 7676114 . A G\n",
      "17 7676114 . A T\n",
      "17 7676120 . C T\n",
      "17 7676123 . C T\n",
      "17 7676135 . T C\n",
      "17 7676147 . G A\n",
      "17 7676153 . G A\n",
      "17 7676154 . G T\n",
      "17 7676159 . A G\n",
      "17 7676165 . C T\n",
      "17 7676168 . T C\n",
      "17 7676189 . T G\n",
      "17 7676220 . A T\n",
      "17 7676222 . A G\n",
      "17 7676224 . C T\n",
      "17 7676228 . C T\n",
      "17 7676246 . A G\n",
      "17 7676267 . G C\n",
      "17 7676269 . G C\n",
      "17 7676269 . G T\n",
      "17 7676275 . G A\n",
      "17 7676395 . T A\n",
      "17 7676529 . T C\n",
      "17 7676532 . G A\n",
      "17 7676535 . T C\n",
      "17 7676553 . C G\n",
      "17 7676565 . G A\n",
      "17 7676587 . T C\n",
      "17 7676589 . C T\n",
      "17 7669689 . G A\n",
      "17 7670622 . T C\n",
      "17 7674187 . T C\n",
      "17 7674260 . T A\n",
      "17 7674942 . C T\n",
      "17 7674980 . A T\n",
      "17 7669655 . C A\n",
      "17 7670684 . C G\n",
      "17 7673555 . C A\n",
      "17 7673564 . G A\n",
      "17 7673696 . C G\n",
      "17 7673716 . C G\n",
      "17 7673752 . G A\n",
      "17 7673769 . G A\n",
      "17 7673796 . C T\n",
      "17 7673837 . A G\n",
      "17 7674257 . A G\n",
      "17 7674858 . C T\n",
      "17 7674942 . C A\n",
      "17 7674944 . C T\n",
      "17 7674947 . A G\n",
      "17 7674979 . C G\n",
      "17 7675162 . T C\n",
      "17 7675224 . G A\n",
      "17 7675995 . G T\n",
      "17 7676099 . G A\n",
      "17 7676201 . T C\n",
      "17 7676220 . A G\n",
      "17 7676388 . G A\n",
      "17 7676568 . G A\n",
      "17 7670677 . C T\n",
      "17 7670686 . G A\n",
      "17 7673596 . T G\n",
      "17 7674231 . G A\n",
      "17 7674263 . A C\n",
      "17 7674326 . C G\n",
      "17 7675092 . T C\n",
      "17 7675108 . G A\n",
      "17 7675232 . G C\n",
      "17 7676162 . A G\n",
      "17 7676278 . G A\n",
      "17 7673394 . C T\n",
      "17 7676167 . C A\n",
      "17 7675095 . C T\n",
      "17 7669671 . C G\n",
      "17 7669676 . T C\n",
      "17 7669691 . C T\n",
      "17 7670610 . T C\n",
      "17 7670621 . C T\n",
      "17 7670630 . C T\n",
      "17 7670636 . T A\n",
      "17 7670643 . C G\n",
      "17 7670649 . G T\n",
      "17 7670684 . C T\n",
      "17 7670716 . C G\n",
      "17 7673552 . C A\n",
      "17 7673564 . G T\n",
      "17 7673579 . G T\n",
      "17 7673582 . G T\n",
      "17 7673598 . G T\n",
      "17 7673698 . T C\n",
      "17 7673712 . C T\n",
      "17 7673730 . T C\n",
      "17 7673743 . C T\n",
      "17 7673753 . G A\n",
      "17 7673787 . G A\n",
      "17 7673793 . G T\n",
      "17 7673802 . C G\n",
      "17 7673805 . A C\n",
      "17 7673823 . C A\n",
      "17 7674179 . A C\n",
      "17 7674184 . G T\n",
      "17 7674194 . G A\n",
      "17 7674199 . A T\n",
      "17 7674220 . C A\n",
      "17 7674248 . T C\n",
      "17 7674295 . G C\n",
      "17 7674863 . G A\n",
      "17 7674891 . G T\n",
      "17 7674893 . C G\n",
      "17 7674927 . G C\n",
      "17 7674944 . C G\n",
      "17 7674945 . G C\n",
      "17 7674954 . G A\n",
      "17 7674962 . G C\n",
      "17 7674963 . G T\n",
      "17 7675054 . A T\n",
      "17 7675069 . G A\n",
      "17 7675071 . G T\n",
      "17 7675075 . A G\n",
      "17 7675132 . C T\n",
      "17 7675153 . G A\n",
      "17 7675156 . C T\n",
      "17 7675201 . C G\n",
      "17 7675211 . A C\n",
      "17 7675999 . A C\n",
      "17 7676018 . C T\n",
      "17 7676033 . G T\n",
      "17 7676040 . C G\n",
      "17 7676048 . G A\n",
      "17 7676072 . G A\n",
      "17 7676084 . A C\n",
      "17 7676087 . T C\n",
      "17 7676092 . G A\n",
      "17 7676093 . G A\n",
      "17 7676096 . C T\n",
      "17 7676105 . G A\n",
      "17 7676136 . G C\n",
      "17 7676155 . G T\n",
      "17 7676163 . G C\n",
      "17 7676181 . G C\n",
      "17 7676182 . C T\n",
      "17 7676211 . C T\n",
      "17 7676223 . T C\n",
      "17 7676225 . G A\n",
      "17 7676233 . A G\n",
      "17 7676387 . C A\n",
      "17 7676537 . A G\n",
      "17 7676541 . T C\n",
      "17 7676567 . C G\n",
      "17 7676572 . G A\n",
      "17 7676581 . T C\n",
      "17 7676583 . C T\n",
      "17 7670601 . T C\n",
      "17 7670700 . G C\n",
      "17 7673775 . C T\n",
      "17 7674210 . G A\n",
      "17 7674220 . C G\n",
      "17 7674291 . C A\n",
      "17 7674879 . C T\n",
      "17 7675151 . C T\n",
      "17 7675994 . C A\n",
      "17 7676066 . T A\n",
      "17 7676090 . C A\n",
      "17 7676157 . G C\n",
      "17 7676214 . T C\n",
      "17 7676276 . T A\n",
      "17 7676557 . G A\n",
      "17 7676584 . G A\n",
      "17 7669593 . C A\n",
      "17 7669655 . C T\n",
      "17 7673547 . A C\n",
      "17 7673826 . A G\n",
      "17 7674265 . T C\n",
      "17 7674876 . G A\n",
      "17 7675074 . C T\n",
      "17 7675089 . G A\n",
      "17 7675122 . T C\n",
      "17 7675139 . C G\n",
      "17 7675219 . G T\n",
      "17 7676235 . A G\n",
      "17 7676274 . T C\n",
      "17 7687390 . G A\n",
      "17 7670626 . C T\n",
      "17 7674797 . T C\n",
      "17 7675045 . C T\n",
      "17 7676301 . G T\n",
      "17 7674229 . C G\n",
      "17 7674888 . T C\n",
      "17 7675236 . A C\n",
      "17 7668819 . C T\n",
      "17 7668855 . G A\n",
      "17 7668862 . G A\n",
      "17 7669262 . A G\n",
      "17 7669510 . C T\n",
      "17 7668632 . C G\n",
      "17 7668783 . C T\n",
      "17 7668836 . G A\n",
      "17 7668996 . G T\n",
      "17 7669124 . C T\n",
      "17 7669282 . G A\n",
      "17 7668673 . T C\n",
      "17 7668874 . G T\n",
      "17 7668983 . G A\n",
      "17 7669061 . C A\n",
      "17 7669263 . C A\n",
      "17 7669404 . C T\n",
      "17 7668449 . A C\n",
      "17 7668539 . G A\n",
      "17 7668803 . G T\n",
      "17 7669044 . C A\n",
      "17 7669227 . G A\n",
      "17 7668388 . C G\n",
      "17 7673291 . C T\n",
      "17 7673301 . C T\n",
      "17 7673308 . T G\n",
      "17 7674327 . G A\n",
      "17 7675977 . C T\n",
      "17 7676300 . A T\n",
      "17 7676507 . A G\n",
      "17 7676512 . T G\n",
      "17 7674232 . C T\n",
      "17 7674217 . C G\n",
      "17 7675094 . A C\n",
      "17 7675094 . A G\n",
      "17 7675208 . C A\n",
      "17 7675209 . A G\n",
      "17 7675207 . G C\n",
      "17 7675209 . A T\n",
      "17 7675209 . A C\n",
      "17 7675189 . G C\n",
      "17 7675191 . A G\n",
      "17 7675190 . C A\n",
      "17 7675085 . C A\n",
      "17 7675086 . A T\n",
      "17 7675086 . A C\n",
      "17 7675084 . G C\n",
      "17 7675086 . A G\n",
      "17 7674250 . C A\n",
      "17 7674250 . C G\n",
      "17 7674251 . A G\n",
      "17 7674251 . A C\n",
      "17 7674238 . C A\n",
      "17 7674239 . A T\n",
      "17 7674237 . G C\n",
      "17 7674239 . A C\n",
      "17 7673796 . C A\n",
      "17 7673796 . C G\n",
      "17 7673797 . A G\n",
      "17 7673779 . C A\n",
      "17 7673779 . C T\n",
      "17 7673777 . G C\n",
      "17 7673779 . C G\n",
      "17 7673778 . T G\n",
      "17 7673763 . T C\n",
      "17 7673764 . C G\n",
      "17 7673763 . T A\n",
      "17 7673763 . T G\n",
      "17 7673811 . A G\n",
      "17 7673812 . A T\n",
      "17 7673811 . A C\n",
      "17 7673812 . A C\n",
      "17 7674233 . C A\n",
      "17 7674233 . C T\n",
      "17 7674232 . C A\n",
      "17 7674233 . C G\n",
      "17 7674229 . C A\n",
      "17 7674230 . C G\n",
      "17 7673824 . C G\n",
      "17 7675076 . T C\n",
      "17 7675075 . A C\n",
      "17 7675076 . T A\n",
      "17 7675077 . G T\n",
      "17 7675077 . G C\n",
      "17 7675076 . T G\n",
      "17 7674953 . T G\n",
      "17 7674954 . G C\n",
      "17 7674954 . G T\n",
      "17 7674890 . T C\n",
      "17 7674890 . T A\n",
      "17 7674948 . T A\n",
      "17 7674947 . A T\n",
      "17 7674947 . A C\n",
      "17 7674946 . G C\n",
      "17 7674200 . T A\n",
      "17 7674199 . A C\n",
      "17 7674199 . A G\n",
      "17 7675216 . C G\n",
      "17 7675217 . T C\n",
      "17 7675218 . T C\n",
      "17 7675217 . T G\n",
      "17 7675218 . T G\n",
      "17 7675217 . T A\n",
      "17 7676037 . A G\n",
      "17 7676037 . A C\n",
      "17 7676037 . A T\n",
      "17 7674950 . A C\n",
      "17 7674950 . A T\n",
      "17 7674950 . A G\n",
      "17 7674253 . A T\n",
      "17 7674247 . T C\n",
      "17 7674247 . T G\n",
      "17 7675160 . G T\n",
      "17 7675160 . G C\n",
      "17 7675161 . G C\n",
      "17 7673788 . G A\n",
      "17 7673788 . G T\n",
      "17 7673787 . G C\n",
      "17 7673788 . G C\n",
      "17 7673787 . G T\n",
      "17 7675181 . T A\n",
      "17 7675180 . C A\n",
      "17 7675089 . G C\n",
      "17 7674893 . C A\n",
      "17 7674894 . G C\n",
      "17 7674221 . G C\n",
      "17 7674217 . C A\n",
      "17 7674218 . T C\n",
      "17 7673802 . C A\n",
      "17 7673803 . G T\n",
      "17 7673781 . C T\n",
      "17 7673782 . T C\n",
      "17 7673775 . C G\n",
      "17 7674887 . C A\n",
      "17 7674886 . A C\n",
      "17 7674887 . C T\n",
      "17 7674241 . G T\n",
      "17 7674242 . A G\n",
      "17 7674242 . A C\n",
      "17 7675996 . T G\n",
      "17 7675995 . G C\n",
      "17 7675095 . C A\n",
      "17 7675094 . A T\n",
      "17 7674885 . C A\n",
      "17 7674884 . A C\n",
      "17 7674884 . A T\n",
      "17 7673805 . A T\n",
      "17 7673800 . C A\n",
      "17 7673799 . A T\n",
      "17 7673799 . A C\n",
      "17 7673799 . A G\n",
      "17 7673800 . C G\n",
      "17 7675125 . A T\n",
      "17 7675125 . A G\n",
      "17 7674917 . T C\n",
      "17 7674917 . T G\n",
      "17 7674918 . A T\n",
      "17 7674918 . A G\n",
      "17 7674918 . A C\n",
      "17 7674873 . A G\n",
      "17 7674873 . A T\n",
      "17 7674873 . A C\n",
      "17 7674262 . T G\n",
      "17 7674263 . A G\n",
      "17 7674263 . A T\n",
      "17 7674256 . T C\n",
      "17 7674257 . A T\n",
      "17 7669596 . G A\n",
      "17 7669603 . A T\n",
      "17 7670593 . G A\n",
      "17 7670606 . C G\n",
      "17 7673534 . C G\n",
      "17 7674171 . G A\n",
      "17 7674309 . C T\n",
      "17 7675129 . G T\n",
      "17 7675135 . G A\n",
      "17 7675141 . G T\n",
      "17 7676102 . G T\n",
      "17 7676605 . C T\n",
      "17 7670644 . A G\n",
      "17 7670727 . G A\n",
      "17 7673595 . G A\n",
      "17 7673789 . A G\n",
      "17 7674246 . G C\n",
      "17 7674844 . A C\n",
      "17 7675255 . G A\n",
      "17 7676283 . G A\n",
      "17 7676289 . A G\n",
      "17 7676417 . G A\n",
      "17 7673621 . C G\n",
      "17 7673622 . A G\n",
      "17 7673692 . C T\n",
      "17 7674291 . C T\n",
      "17 7674850 . A G\n",
      "17 7674967 . C T\n",
      "17 7675253 . G A\n",
      "17 7675978 . C T\n",
      "17 7675981 . C T\n",
      "17 7676606 . G A\n",
      "17 7676610 . G A\n",
      "17 7676614 . G C\n",
      "17 7669594 . A G\n",
      "17 7669693 . G A\n",
      "17 7673544 . G A\n",
      "17 7673619 . G A\n",
      "17 7673684 . T C\n",
      "17 7673686 . G C\n",
      "17 7673694 . T G\n",
      "17 7673720 . G C\n",
      "17 7673740 . C A\n",
      "17 7676044 . A C\n",
      "17 7676213 . T C\n",
      "17 7676367 . A G\n",
      "17 7669657 . G A\n",
      "17 7673527 . C T\n",
      "17 7673609 . C G\n",
      "17 7673841 . C T\n",
      "17 7673577 . C A\n",
      "17 7673596 . T C\n",
      "17 7674227 . T A\n",
      "17 7674868 . C T\n",
      "17 7673715 . C T\n",
      "17 7674957 . G A\n",
      "17 7673839 . T C\n",
      "17 7674190 . T C\n",
      "17 7674203 . T C\n",
      "17 7675056 . C T\n",
      "17 7675075 . A T\n",
      "17 7675148 . G T\n",
      "17 7674236 . T C\n",
      "17 7676040 . C A\n",
      "17 7676237 . C T\n",
      "17 7676266 . A T\n",
      "17 7674867 . G A\n",
      "17 7674900 . T C\n",
      "17 7674903 . T A\n",
      "17 7675082 . G A\n",
      "17 7676048 . G T\n",
      "17 7676050 . A T\n",
      "17 7676056 . C G\n",
      "17 7676234 . C T\n",
      "17 7676399 . G T\n",
      "17 7676550 . A G\n",
      "17 7669640 . A G\n",
      "17 7670599 . G T\n",
      "17 7670667 . A C\n",
      "17 7673613 . G T\n",
      "17 7673725 . G A\n",
      "17 7673777 . G T\n",
      "17 7673809 . C T\n",
      "17 7673828 . T C\n",
      "17 7674216 . C G\n",
      "17 7674885 . C G\n",
      "17 7674980 . A C\n",
      "17 7675990 . T C\n",
      "17 7676039 . A G\n",
      "17 7676175 . C G\n",
      "17 7676230 . G T\n",
      "17 7676577 . T C\n",
      "17 7676584 . G C\n",
      "17 7673587 . G A\n",
      "17 7673604 . C G\n",
      "17 7673745 . T C\n",
      "17 7674223 . T C\n",
      "17 7675061 . T C\n",
      "17 7676131 . G A\n",
      "17 7676159 . A T\n",
      "17 7676195 . T C\n",
      "17 7676558 . G A\n",
      "17 7670669 . G A\n",
      "17 7670696 . A G\n",
      "17 7673536 . T C\n",
      "17 7673558 . C G\n",
      "17 7673736 . G A\n",
      "17 7673759 . C G\n",
      "17 7673767 . C T\n",
      "17 7673784 . C T\n",
      "17 7673791 . A G\n",
      "17 7674214 . G A\n",
      "17 7674253 . A C\n",
      "17 7674277 . C T\n",
      "17 7674278 . A G\n",
      "17 7674897 . A T\n",
      "17 7674962 . G A\n",
      "17 7674971 . C T\n",
      "17 7675066 . G C\n",
      "17 7675080 . G T\n",
      "17 7675091 . C T\n",
      "17 7675130 . G T\n",
      "17 7675155 . G A\n",
      "17 7675213 . C T\n",
      "17 7676011 . T G\n",
      "17 7676016 . G A\n",
      "17 7676043 . A G\n",
      "17 7676210 . C T\n",
      "17 7676564 . C T\n",
      "17 7669630 . T C\n",
      "17 7670709 . C A\n",
      "17 7673535 . C T\n",
      "17 7673598 . G C\n",
      "17 7673699 . A C\n",
      "17 7673700 . C G\n",
      "17 7673737 . G A\n",
      "17 7673794 . C G\n",
      "17 7673820 . C G\n",
      "17 7673835 . C A\n",
      "17 7674226 . A G\n",
      "17 7674860 . T G\n",
      "17 7674888 . T G\n",
      "17 7675052 . C T\n",
      "17 7675097 . A C\n",
      "17 7675098 . C A\n",
      "17 7675109 . T C\n",
      "17 7675120 . C G\n",
      "17 7675142 . A G\n",
      "17 7675170 . C A\n",
      "17 7675174 . C T\n",
      "17 7675220 . T A\n",
      "17 7675223 . A G\n",
      "17 7675236 . A T\n",
      "17 7676056 . C T\n",
      "17 7676381 . C G\n",
      "17 7676405 . T C\n",
      "17 7687459 . T C\n",
      "17 7670694 . C A\n",
      "17 7669676 . T G\n",
      "17 7674211 . A G\n",
      "17 7674914 . A G\n",
      "17 7675992 . A G\n",
      "17 7676599 . G A\n",
      "17 7675327 . C T\n",
      "17 7675134 . T C\n",
      "17 7673579 . G A\n",
      "17 7674239 . A G\n",
      "17 7674935 . C A\n",
      "17 7676241 . A T\n",
      "17 7669626 . C T\n",
      "17 7670677 . C G\n",
      "17 7670713 . G C\n",
      "17 7673586 . G T\n",
      "17 7673603 . G A\n",
      "17 7673770 . T A\n",
      "17 7673838 . C T\n",
      "17 7674178 . G T\n",
      "17 7674193 . A G\n",
      "17 7674232 . C G\n",
      "17 7674243 . A C\n",
      "17 7674916 . A G\n",
      "17 7675136 . G A\n",
      "17 7675235 . T C\n",
      "17 7676028 . A G\n",
      "17 7676124 . G C\n",
      "17 7676140 . G T\n",
      "17 7676177 . G C\n",
      "17 7676513 . C T\n",
      "17 7669609 . T A\n",
      "17 7670599 . G A\n",
      "17 7670624 . C T\n",
      "17 7670657 . T C\n",
      "17 7673751 . C G\n",
      "17 7674286 . C A\n",
      "17 7674292 . T C\n",
      "17 7675044 . G C\n",
      "17 7675082 . G C\n",
      "17 7675149 . T C\n",
      "17 7675997 . G T\n",
      "17 7676565 . G C\n",
      "17 7670649 . G A\n",
      "17 7673553 . T C\n",
      "17 7673592 . G C\n",
      "17 7673785 . C T\n",
      "17 7675049 . T C\n",
      "17 7675096 . A C\n",
      "17 7676080 . C A\n",
      "17 7676569 . C T\n",
      "17 7669640 . A C\n",
      "17 7670625 . T A\n",
      "17 7670720 . A G\n",
      "17 7670721 . G T\n",
      "17 7673528 . T C\n",
      "17 7673703 . C T\n",
      "17 7673771 . G C\n",
      "17 7673790 . C A\n",
      "17 7673824 . C T\n",
      "17 7674190 . T G\n",
      "17 7674271 . G A\n",
      "17 7674925 . A G\n",
      "17 7675224 . G C\n",
      "17 7675226 . G C\n",
      "17 7675238 . T A\n",
      "17 7676047 . C G\n",
      "17 7676258 . G A\n",
      "17 7669611 . A C\n",
      "17 7669638 . A G\n",
      "17 7670631 . C A\n",
      "17 7670655 . C A\n",
      "17 7670671 . C T\n",
      "17 7670701 . C T\n",
      "17 7673608 . G C\n",
      "17 7669631 . G C\n",
      "17 7673772 . C G\n",
      "17 7669642 . G T\n",
      "17 7674226 . A T\n",
      "17 7669687 . G C\n",
      "17 7674288 . A G\n",
      "17 7674861 . C T\n",
      "17 7670628 . C G\n",
      "17 7674946 . G T\n",
      "17 7670647 . C T\n",
      "17 7675079 . T G\n",
      "17 7670668 . G C\n",
      "17 7670695 . G C\n",
      "17 7673554 . C T\n",
      "17 7675169 . T G\n",
      "17 7675178 . A T\n",
      "17 7669658 . G C\n",
      "17 7669687 . G A\n",
      "17 7675204 . T A\n",
      "17 7675206 . G C\n",
      "17 7675228 . A G\n",
      "17 7670624 . C A\n",
      "17 7673795 . A C\n",
      "17 7675240 . T A\n",
      "17 7676209 . A G\n",
      "17 7676381 . C A\n",
      "17 7670661 . G A\n",
      "17 7676400 . A G\n",
      "17 7674290 . C T\n",
      "17 7676556 . A C\n",
      "17 7674883 . C T\n",
      "17 7673708 . A G\n",
      "17 7674924 . C G\n",
      "17 7674964 . G A\n",
      "17 7675083 . G T\n",
      "17 7673829 . A G\n",
      "17 7674187 . T A\n",
      "17 7675099 . C T\n",
      "17 7675152 . C G\n",
      "17 7674225 . C T\n",
      "17 7675166 . G A\n",
      "17 7675184 . A C\n",
      "17 7674251 . A T\n",
      "17 7674253 . A G\n",
      "17 7675207 . G A\n",
      "17 7674269 . T G\n",
      "17 7674282 . A C\n",
      "17 7675212 . A T\n",
      "17 7675230 . G T\n",
      "17 7674901 . G A\n",
      "17 7676014 . C G\n",
      "17 7675120 . C T\n",
      "17 7676024 . A G\n",
      "17 7676094 . G A\n",
      "17 7676106 . G A\n",
      "17 7675158 . G A\n",
      "17 7675198 . G A\n",
      "17 7676155 . G A\n",
      "17 7675237 . C T\n",
      "17 7675992 . A C\n",
      "17 7675994 . C G\n",
      "17 7676020 . C T\n",
      "17 7676271 . G T\n",
      "17 7676198 . G A\n",
      "17 7676574 . A C\n",
      "17 7676248 . C T\n",
      "17 7676385 . A C\n",
      "17 7669668 . G A\n",
      "17 7670671 . C G\n",
      "17 7670705 . C T\n",
      "17 7670717 . T C\n",
      "17 7673530 . A G\n",
      "17 7673555 . C T\n",
      "17 7673601 . G T\n",
      "17 7673606 . G A\n",
      "17 7673609 . C A\n",
      "17 7673611 . A G\n",
      "17 7673718 . G A\n",
      "17 7673749 . T C\n",
      "17 7673822 . T A\n",
      "17 7673831 . A T\n",
      "17 7674191 . C A\n",
      "17 7674210 . G C\n",
      "17 7674246 . G A\n",
      "17 7674294 . G C\n",
      "17 7674924 . C A\n",
      "17 7674942 . C G\n",
      "17 7675064 . G A\n",
      "17 7675078 . G A\n",
      "17 7675080 . G C\n",
      "17 7675119 . G T\n",
      "17 7675142 . A T\n",
      "17 7675227 . C T\n",
      "17 7675989 . C T\n",
      "17 7676001 . G A\n",
      "17 7676026 . G T\n",
      "17 7676125 . G C\n",
      "17 7676137 . C T\n",
      "17 7676171 . C T\n",
      "17 7676176 . T G\n",
      "17 7676384 . G A\n",
      "17 7676398 . G A\n",
      "17 7676557 . G C\n",
      "17 7675112 . T G\n",
      "17 7676167 . C G\n",
      "17 7667908 . C A\n",
      "17 7675103 . G C\n",
      "17 7676591 . C T\n",
      "17 7675048 . C T\n",
      "17 7673248 . A G\n",
      "17 7673250 . C T\n",
      "17 7673307 . T C\n",
      "17 7673319 . T G\n",
      "17 7673336 . G C\n",
      "17 7673520 . C A\n",
      "17 7673625 . G T\n",
      "17 7673872 . T C\n",
      "17 7673895 . C T\n",
      "17 7673896 . C T\n",
      "17 7673897 . C T\n",
      "17 7673899 . T C\n",
      "17 7674195 . T A\n",
      "17 7674843 . C T\n",
      "17 7674844 . A G\n",
      "17 7674845 . G C\n",
      "17 7674986 . T G\n",
      "17 7675256 . A G\n",
      "17 7676370 . C G\n",
      "17 7676415 . G T\n",
      "17 7676457 . C T\n",
      "17 7676465 . G C\n",
      "17 7676499 . C T\n",
      "17 7676522 . G A\n",
      "17 7669664 . G A\n",
      "17 7670632 . T C\n",
      "17 7673222 . C T\n",
      "17 7673251 . G A\n",
      "17 7673287 . G A\n",
      "17 7673312 . A G\n",
      "17 7673575 . G A\n",
      "17 7673903 . A C\n",
      "17 7674840 . C T\n",
      "17 7674880 . C A\n",
      "17 7673223 . G A\n",
      "17 7675114 . T C\n",
      "17 7675210 . A G\n",
      "17 7673225 . C T\n",
      "17 7676421 . A C\n",
      "17 7673322 . T C\n",
      "17 7673337 . G A\n",
      "17 7676459 . C A\n",
      "17 7673346 . A T\n",
      "17 7673765 . C T\n",
      "17 7676477 . T C\n",
      "17 7673774 . C T\n",
      "17 7673870 . A G\n",
      "17 7676484 . G A\n",
      "17 7674167 . A C\n",
      "17 7674293 . A G\n",
      "17 7674295 . G T\n",
      "17 7675033 . C T\n",
      "17 7676443 . C T\n",
      "17 7676444 . G A\n",
      "17 7676445 . C A\n",
      "17 7676452 . C A\n",
      "17 7676486 . G A\n",
      "17 7676494 . G C\n",
      "17 7669614 . C T\n",
      "17 7670594 . G A\n",
      "17 7673226 . G A\n",
      "17 7673228 . G C\n",
      "17 7673245 . A G\n",
      "17 7673290 . C T\n",
      "17 7673291 . C G\n",
      "17 7673848 . T C\n",
      "17 7673889 . A C\n",
      "17 7674303 . A G\n",
      "17 7676035 . C T\n",
      "17 7676258 . G C\n",
      "17 7676280 . T A\n",
      "17 7676420 . G T\n",
      "17 7676430 . C T\n",
      "17 7676462 . C G\n",
      "17 7676480 . G A\n",
      "17 7676483 . G T\n",
      "17 7673789 . A T\n",
      "17 7674972 . C G\n",
      "17 7675029 . G A\n",
      "17 7676051 . G C\n",
      "17 7676368 . A G\n",
      "17 7673700 . C A\n",
      "17 7676568 . G C\n",
      "17 7674180 . C A\n",
      "17 7673755 . G C\n",
      "17 7674205 . G T\n",
      "17 7674857 . A T\n",
      "17 7673531 . G A\n",
      "17 7675040 . C T\n",
      "17 7675162 . T G\n",
      "17 7674175 . C T\n",
      "17 7675243 . G A\n",
      "17 7676397 . A G\n",
      "17 7670698 . G A\n",
      "17 7675093 . C T\n",
      "17 7676156 . G A\n",
      "17 7676636 . A C\n",
      "17 7669660 . G T\n",
      "17 7673750 . G T\n",
      "17 7674285 . G T\n",
      "17 7676093 . G T\n",
      "17 7676586 . C T\n",
      "17 7687368 . G A\n",
      "17 7670645 . G A\n",
      "17 7673581 . G A\n",
      "17 7673616 . A G\n",
      "17 7674900 . T G\n",
      "17 7675113 . G A\n",
      "17 7675115 . G A\n",
      "17 7675139 . C A\n",
      "17 7675173 . C T\n",
      "17 7675199 . G T\n",
      "17 7675992 . A T\n",
      "17 7676140 . G A\n",
      "17 7676163 . G A\n",
      "17 7673693 . T G\n",
      "17 7673704 . G T\n",
      "17 7673833 . T G\n",
      "17 7676411 . A G\n",
      "17 7673597 . T G\n",
      "17 7673612 . G C\n",
      "17 7674889 . A G\n",
      "17 7674966 . C A\n",
      "17 7675097 . A G\n",
      "17 7675215 . T C\n",
      "17 7676076 . G A\n",
      "17 7676169 . G T\n",
      "17 7670602 . C T\n",
      "17 7670613 . A G\n",
      "17 7673534 . C T\n",
      "17 7673726 . C T\n",
      "17 7674197 . T G\n",
      "17 7674211 . A T\n",
      "17 7674878 . A C\n",
      "17 7674977 . A G\n",
      "17 7676100 . G A\n",
      "17 7676399 . G A\n",
      "17 7676560 . G A\n",
      "17 7676570 . T G\n",
      "17 7673850 . G C\n",
      "17 7676287 . A G\n",
      "17 7673831 . A G\n",
      "17 7669645 . T C\n",
      "17 7673773 . G C\n",
      "17 7674178 . G A\n",
      "17 7675078 . G T\n",
      "17 7675170 . C T\n",
      "17 7676073 . G C\n",
      "17 7676259 . G A\n",
      "17 7676560 . G T\n",
      "17 7674935 . C T\n",
      "17 7676127 . G A\n",
      "17 7676164 . C T\n",
      "17 7676218 . C A\n",
      "17 7676253 . G A\n",
      "17 7676377 . C T\n",
      "17 7673578 . T C\n",
      "17 7673845 . G C\n",
      "17 7674869 . T G\n",
      "17 7674940 . C T\n",
      "17 7675106 . A C\n",
      "17 7675205 . T G\n",
      "17 7676059 . G A\n",
      "17 7676065 . T A\n",
      "17 7676073 . G A\n",
      "17 7676150 . C T\n",
      "17 7676158 . G A\n",
      "17 7676158 . G C\n",
      "17 7676217 . T C\n",
      "17 7669662 . T C\n",
      "17 7669673 . T C\n",
      "17 7670618 . G A\n",
      "17 7673805 . A G\n",
      "17 7673806 . C G\n",
      "17 7674277 . C G\n",
      "17 7675234 . G C\n",
      "17 7675989 . C G\n",
      "17 7676555 . G C\n",
      "17 7670640 . T C\n",
      "17 7673232 . A T\n",
      "17 7674245 . T C\n",
      "17 7674908 . T C\n",
      "17 7675195 . C A\n",
      "17 7676185 . C A\n",
      "17 7669644 . G A\n",
      "17 7673801 . A G\n",
      "17 7675123 . G C\n",
      "17 7674853 . C A\n",
      "17 7669618 . G C\n",
      "17 7669666 . C T\n",
      "17 7669672 . C T\n",
      "17 7669681 . C T\n",
      "17 7670648 . T C\n",
      "17 7670696 . A C\n",
      "17 7673558 . C A\n",
      "17 7673577 . C T\n",
      "17 7673747 . C G\n",
      "17 7673752 . G C\n",
      "17 7674886 . A G\n",
      "17 7674906 . T C\n",
      "17 7676006 . A C\n",
      "17 7676052 . C A\n",
      "17 7676107 . C G\n",
      "17 7676160 . G A\n",
      "17 7676268 . G C\n",
      "17 7673358 . G A\n",
      "17 7674299 . T C\n",
      "17 7674985 . G T\n",
      "17 7676434 . T C\n",
      "17 7676487 . G T\n",
      "17 7670597 . G C\n",
      "17 7673204 . T C\n",
      "17 7673217 . A G\n",
      "17 7673225 . C A\n",
      "17 7673246 . C T\n",
      "17 7673303 . C G\n",
      "17 7673901 . T G\n",
      "17 7673218 . C T\n",
      "17 7673359 . G A\n",
      "17 7673871 . G A\n",
      "17 7676467 . G A\n",
      "17 7673884 . G C\n",
      "17 7673893 . G A\n",
      "17 7674858 . C A\n",
      "17 7673895 . C A\n",
      "17 7675976 . C T\n",
      "17 7674983 . C T\n",
      "17 7676433 . A G\n",
      "17 7676435 . G A\n",
      "17 7676446 . T G\n",
      "17 7676450 . C T\n",
      "17 7676489 . G C\n",
      "17 7670698 . G C\n",
      "17 7673751 . C A\n",
      "17 7673696 . C A\n",
      "17 7673700 . C T\n",
      "17 7673699 . A T\n",
      "17 7670664 . C A\n",
      "17 7670673 . C A\n",
      "17 7673764 . C A\n",
      "17 7673780 . T G\n",
      "17 7673803 . G C\n",
      "17 7673812 . A G\n",
      "17 7674188 . C A\n",
      "17 7674252 . C A\n",
      "17 7674268 . A C\n",
      "17 7674870 . C A\n",
      "17 7674871 . A T\n",
      "17 7674936 . C A\n",
      "17 7674939 . C A\n",
      "17 7674941 . A T\n",
      "17 7675064 . G C\n",
      "17 7675074 . C A\n",
      "17 7675101 . C A\n",
      "17 7675115 . G C\n",
      "17 7675145 . C G\n",
      "17 7675175 . C T\n",
      "17 7675182 . G A\n",
      "17 7675197 . T A\n",
      "17 7675216 . C A\n",
      "17 7676028 . A T\n",
      "17 7676031 . A C\n",
      "17 7676043 . A C\n",
      "17 7676055 . C A\n",
      "17 7676071 . G A\n",
      "17 7676176 . T A\n",
      "17 7673534 . C A\n",
      "17 7673838 . C A\n",
      "17 7674180 . C T\n",
      "17 7674972 . C T\n",
      "17 7674974 . A C\n",
      "17 7675989 . C A\n",
      "17 7675993 . C T\n",
      "17 7673533 . A C\n",
      "17 7673839 . T A\n",
      "17 7673839 . T G\n",
      "17 7674292 . T A\n",
      "17 7674857 . A C\n",
      "17 7674859 . C A\n",
      "17 7675051 . A T\n",
      "17 7675053 . C G\n",
      "17 7675993 . C A\n",
      "17 7669611 . A T\n",
      "17 7669639 . C T\n",
      "17 7669671 . C A\n",
      "17 7669687 . G T\n",
      "17 7670664 . C G\n",
      "17 7670670 . C T\n",
      "17 7670673 . C T\n",
      "17 7673539 . A G\n",
      "17 7673542 . G A\n",
      "17 7673545 . A G\n",
      "17 7673600 . T C\n",
      "17 7673608 . G A\n",
      "17 7673729 . G T\n",
      "17 7673790 . C G\n",
      "17 7673815 . T G\n",
      "17 7673830 . G T\n",
      "17 7674202 . A C\n",
      "17 7674238 . C G\n",
      "17 7674287 . C T\n",
      "17 7674890 . T G\n",
      "17 7674923 . A T\n",
      "17 7674963 . G A\n",
      "17 7675083 . G A\n",
      "17 7675112 . T C\n",
      "17 7675151 . C A\n",
      "17 7675154 . G A\n",
      "17 7675194 . T C\n",
      "17 7675197 . T C\n",
      "17 7675232 . G T\n",
      "17 7676023 . A C\n",
      "17 7676050 . A C\n",
      "17 7676068 . T G\n",
      "17 7676112 . G A\n",
      "17 7676133 . G C\n",
      "17 7676232 . G T\n",
      "17 7676236 . G C\n",
      "17 7676242 . A T\n",
      "17 7676251 . T C\n",
      "17 7676271 . G A\n",
      "17 7676382 . C T\n",
      "17 7676391 . G C\n",
      "17 7676398 . G C\n",
      "17 7676547 . C G\n",
      "17 7676560 . G C\n",
      "17 7676573 . G T\n",
      "17 7675048 . C G\n",
      "17 7676273 . C T\n",
      "17 7670604 . C T\n",
      "17 7673530 . A C\n",
      "17 7675047 . G T\n",
      "17 7675993 . C G\n",
      "17 7676273 . C A\n",
      "17 7673844 . A C\n",
      "17 7674855 . A G\n",
      "17 7675052 . C A\n",
      "17 7676283 . G C\n",
      "17 7673777 . G A\n",
      "17 7674639 . C A\n",
      "17 7675322 . A G\n",
      "17 7676799 . C G\n",
      "17 7675353 . C T\n",
      "17 7675519 . A G\n",
      "17 7676038 . G A\n",
      "17 7676243 . A G\n",
      "17 7669660 . G A\n",
      "17 7673580 . G A\n",
      "17 7673586 . G C\n",
      "17 7673847 . A G\n",
      "17 7676282 . G A\n",
      "17 7669624 . C A\n",
      "17 7673753 . G C\n",
      "17 7674915 . A G\n",
      "17 7675063 . T C\n",
      "17 7676033 . G C\n",
      "17 7676069 . C T\n",
      "17 7676535 . T A\n",
      "17 7676538 . A G\n",
      "17 7669697 . G A\n",
      "17 7676514 . C A\n",
      "17 7674173 . C A\n",
      "17 7674975 . A G\n",
      "17 7675986 . C A\n",
      "17 7676276 . T C\n",
      "17 7676374 . G C\n",
      "17 7670713 . G A\n",
      "17 7673720 . G A\n",
      "17 7673768 . T C\n",
      "17 7674255 . G A\n",
      "17 7676057 . C T\n",
      "17 7676087 . T A\n",
      "17 7676126 . T C\n",
      "17 7676174 . T C\n",
      "17 7676252 . T C\n",
      "17 7673618 . A G\n",
      "17 7687372 . C A\n",
      "17 7687391 . C A\n",
      "17 7675184 . A G\n",
      "17 7670579 . T A\n",
      "17 7669614 . C G\n",
      "17 7669623 . G T\n",
      "17 7669634 . T C\n",
      "17 7669661 . G A\n",
      "17 7670620 . C T\n",
      "17 7670656 . C A\n",
      "17 7670679 . G A\n",
      "17 7670690 . A T\n",
      "17 7670693 . T C\n",
      "17 7673537 . G T\n",
      "17 7673560 . A G\n",
      "17 7673582 . G A\n",
      "17 7673707 . T A\n",
      "17 7673741 . C T\n",
      "17 7673755 . G A\n",
      "17 7673794 . C T\n",
      "17 7673816 . G A\n",
      "17 7673818 . T A\n",
      "17 7673819 . C T\n",
      "17 7673825 . C A\n",
      "17 7673832 . T C\n",
      "17 7674194 . G C\n",
      "17 7674206 . T C\n",
      "17 7674207 . G T\n",
      "17 7674245 . T G\n",
      "17 7674860 . T C\n",
      "17 7674875 . G A\n",
      "17 7674883 . C G\n",
      "17 7674894 . G T\n",
      "17 7674898 . A C\n",
      "17 7674909 . C G\n",
      "17 7674926 . C G\n",
      "17 7674941 . A C\n",
      "17 7674949 . A T\n",
      "17 7674952 . A T\n",
      "17 7675070 . C G\n",
      "17 7675081 . G C\n",
      "17 7675098 . C T\n",
      "17 7675102 . C A\n",
      "17 7675156 . C G\n",
      "17 7675157 . G C\n",
      "17 7675174 . C A\n",
      "17 7675188 . G A\n",
      "17 7675223 . A T\n",
      "17 7675227 . C A\n",
      "17 7675229 . G A\n",
      "17 7675233 . A T\n",
      "17 7676031 . A G\n",
      "17 7676041 . G C\n",
      "17 7676060 . G C\n",
      "17 7676066 . T C\n",
      "17 7676077 . G A\n",
      "17 7676104 . G T\n",
      "17 7676176 . T C\n",
      "17 7676186 . A T\n",
      "17 7676190 . G A\n",
      "17 7676193 . C T\n",
      "17 7676229 . G C\n",
      "17 7676262 . G A\n",
      "17 7676267 . G A\n",
      "17 7676268 . G A\n",
      "17 7676393 . T C\n",
      "17 7676536 . G A\n",
      "17 7676553 . C T\n",
      "17 7676566 . A C\n",
      "17 7676568 . G T\n",
      "17 7673612 . G A\n",
      "17 7669646 . T G\n",
      "17 7669672 . C A\n",
      "17 7670609 . C G\n",
      "17 7670619 . C A\n",
      "17 7670661 . G C\n",
      "17 7670664 . C T\n",
      "17 7670705 . C G\n",
      "17 7673567 . T A\n",
      "17 7673578 . T A\n",
      "17 7673587 . G C\n",
      "17 7673807 . C A\n",
      "17 7673820 . C A\n",
      "17 7674196 . G A\n",
      "17 7674202 . A T\n",
      "17 7674205 . G A\n",
      "17 7674234 . C T\n",
      "17 7674268 . A T\n",
      "17 7674272 . T C\n",
      "17 7674289 . A T\n",
      "17 7674916 . A C\n",
      "17 7674959 . G T\n",
      "17 7675053 . C T\n",
      "17 7675097 . A T\n",
      "17 7675107 . T C\n",
      "17 7675140 . G C\n",
      "17 7675172 . A G\n",
      "17 7675187 . G A\n",
      "17 7676008 . A G\n",
      "17 7676019 . C A\n",
      "17 7676038 . G T\n",
      "17 7676098 . A G\n",
      "17 7676137 . C A\n",
      "17 7676145 . G A\n",
      "17 7676154 . G A\n",
      "17 7676170 . G A\n",
      "17 7676185 . C T\n",
      "17 7676218 . C T\n",
      "17 7676240 . C A\n",
      "17 7676247 . T C\n",
      "17 7676392 . T G\n",
      "17 7676548 . T C\n",
      "17 7676588 . C G\n",
      "17 7676588 . C T\n",
      "17 7673617 . G T\n",
      "17 7673843 . C T\n",
      "17 7674853 . C T\n",
      "17 7676381 . C T\n",
      "17 7675210 . A T\n",
      "17 7668477 . G A\n",
      "17 7668848 . A G\n",
      "17 7668849 . A G\n",
      "17 7669200 . G T\n",
      "17 7669252 . A G\n",
      "17 7669281 . C T\n",
      "17 7669441 . G A\n",
      "17 7669465 . C T\n",
      "17 7676616 . C T\n",
      "17 7687471 . G A\n",
      "17 7675142 . A C\n",
      "17 7676010 . T A\n",
      "17 7667874 . G C\n",
      "17 7667880 . G A\n",
      "17 7667888 . G A\n",
      "17 7667899 . A C\n",
      "17 7667901 . C T\n",
      "17 7667904 . C T\n",
      "17 7667906 . C A\n",
      "17 7667906 . C T\n",
      "17 7667909 . C T\n",
      "17 7667910 . C A\n",
      "17 7667914 . C T\n",
      "17 7667923 . T G\n",
      "17 7667926 . C T\n",
      "17 7667929 . C T\n",
      "17 7667933 . C T\n",
      "17 7669678 . G A\n",
      "17 7669689 . G T\n",
      "17 7670612 . G C\n",
      "17 7673537 . G A\n",
      "17 7673584 . G C\n",
      "17 7673727 . T C\n",
      "17 7673729 . G C\n",
      "17 7673744 . T C\n",
      "17 7673759 . C A\n",
      "17 7673766 . T C\n",
      "17 7673782 . T G\n",
      "17 7674201 . G A\n",
      "17 7674221 . G T\n",
      "17 7674260 . T C\n",
      "17 7674270 . G T\n",
      "17 7674271 . G C\n",
      "17 7674279 . G C\n",
      "17 7674912 . C G\n",
      "17 7674932 . T C\n",
      "17 7674955 . C A\n",
      "17 7675105 . C A\n",
      "17 7675219 . G A\n",
      "17 7675234 . G A\n",
      "17 7676116 . G A\n",
      "17 7676150 . C A\n",
      "17 7676164 . C G\n",
      "17 7676238 . A C\n",
      "17 7676402 . G T\n",
      "17 7676590 . T C\n",
      "17 7676605 . C A\n",
      "17 7673249 . T C\n",
      "17 7673351 . A G\n",
      "17 7673531 . G C\n",
      "17 7676462 . C T\n",
      "17 7676473 . G A\n",
      "17 7676475 . G T\n",
      "17 7673236 . G A\n",
      "17 7673258 . A T\n",
      "17 7673295 . T A\n",
      "17 7673350 . T C\n",
      "17 7673850 . G A\n",
      "17 7673851 . G A\n",
      "17 7674295 . G A\n",
      "17 7669703 . G A\n",
      "17 7670589 . A C\n",
      "17 7673240 . G A\n",
      "17 7673332 . G A\n",
      "17 7673340 . C T\n",
      "17 7673346 . A G\n",
      "17 7673918 . T A\n",
      "17 7675249 . G A\n",
      "17 7675250 . A C\n",
      "17 7676491 . T C\n",
      "17 7676493 . G C\n",
      "17 7676509 . G A\n",
      "17 7673268 . T C\n",
      "17 7673883 . G A\n",
      "17 7676275 . G C\n",
      "17 7676497 . T G\n",
      "17 7670620 . C A\n",
      "17 7670708 . C T\n",
      "17 7674286 . C G\n",
      "17 7674920 . T C\n",
      "17 7675078 . G C\n",
      "17 7675128 . T A\n",
      "17 7676013 . G C\n",
      "17 7676058 . T A\n",
      "17 7676187 . T C\n",
      "17 7676197 . G T\n",
      "17 7673543 . T C\n",
      "17 7674969 . G C\n",
      "17 7675122 . T A\n",
      "17 7675233 . A G\n",
      "17 7676109 . G A\n",
      "17 7676256 . T G\n",
      "17 7676269 . G A\n",
      "17 7676380 . A T\n",
      "17 7670679 . G C\n",
      "17 7670693 . T A\n",
      "17 7673767 . C G\n",
      "17 7673797 . A T\n",
      "17 7673824 . C A\n",
      "17 7673826 . A T\n",
      "17 7674242 . A T\n",
      "17 7674275 . T C\n",
      "17 7674279 . G T\n",
      "17 7674875 . G C\n",
      "17 7674905 . C T\n",
      "17 7675082 . G T\n",
      "17 7673770 . T C\n",
      "17 7674225 . C A\n",
      "17 7675092 . T A\n",
      "17 7676567 . C A\n",
      "17 7674974 . A G\n",
      "17 7669662 . T G\n",
      "17 7669650 . T C\n",
      "17 7674921 . C A\n",
      "17 7676097 . C T\n",
      "17 7673780 . T A\n",
      "17 7674955 . C G\n",
      "17 7675212 . A C\n",
      "17 7676604 . C A\n",
      "17 7673584 . G A\n",
      "17 7673740 . C T\n",
      "17 7673746 . T C\n",
      "17 7673825 . C T\n",
      "17 7675148 . G A\n",
      "17 7675203 . G C\n",
      "17 7675239 . G C\n",
      "17 7676029 . A C\n",
      "17 7673795 . A T\n",
      "17 7669612 . G C\n",
      "17 7670682 . C G\n",
      "17 7670689 . C T\n",
      "17 7670700 . G T\n",
      "17 7673540 . G A\n",
      "17 7673550 . T A\n",
      "17 7673696 . C T\n",
      "17 7673709 . G A\n",
      "17 7673711 . G C\n",
      "17 7673745 . T G\n",
      "17 7673758 . T C\n",
      "17 7674278 . A C\n",
      "17 7674899 . G C\n",
      "17 7674936 . C T\n",
      "17 7674960 . G A\n",
      "17 7674963 . G C\n",
      "17 7675050 . C T\n",
      "17 7675136 . G T\n",
      "17 7675178 . A C\n",
      "17 7675210 . A C\n",
      "17 7675998 . C G\n",
      "17 7676080 . C T\n",
      "17 7676125 . G A\n",
      "17 7676376 . C T\n",
      "17 7676552 . T G\n",
      "17 7670606 . C T\n",
      "17 7670645 . G C\n",
      "17 7673716 . C T\n",
      "17 7674244 . C A\n",
      "17 7674908 . T A\n",
      "17 7674956 . T G\n",
      "17 7676077 . G C\n",
      "17 7676254 . C T\n",
      "17 7676272 . A T\n",
      "17 7676518 . C T\n",
      "17 7669673 . T A\n",
      "17 7670649 . G C\n",
      "17 7670684 . C A\n",
      "17 7673749 . T G\n",
      "17 7674176 . C T\n",
      "17 7674209 . G A\n",
      "17 7674856 . G A\n",
      "17 7674891 . G A\n",
      "17 7674933 . T C\n",
      "17 7675047 . G A\n",
      "17 7675090 . C G\n",
      "17 7675157 . G T\n",
      "17 7675200 . C T\n",
      "17 7675220 . T C\n",
      "17 7675235 . T G\n",
      "17 7676019 . C G\n",
      "17 7676043 . A T\n",
      "17 7676068 . T C\n",
      "17 7675140 . G T\n",
      "17 7676404 . C T\n",
      "17 7673570 . T A\n",
      "17 7674261 . G C\n",
      "17 7676048 . G C\n",
      "17 7670704 . A C\n",
      "17 7670721 . G A\n",
      "17 7673792 . G A\n",
      "17 7674258 . G A\n",
      "17 7675245 . A T\n",
      "17 7676090 . C T\n",
      "17 7676180 . A G\n",
      "17 7676228 . C A\n",
      "17 7676555 . G A\n",
      "17 7676583 . C A\n",
      "17 7670668 . G A\n",
      "17 7670725 . C G\n",
      "17 7674231 . G T\n",
      "17 7674264 . G A\n",
      "17 7674880 . C G\n",
      "17 7674961 . A C\n",
      "17 7676015 . T C\n",
      "17 7676372 . T C\n",
      "17 7676511 . A C\n",
      "17 7669684 . C T\n",
      "17 7669698 . A T\n",
      "17 7670600 . G T\n",
      "17 7670614 . G A\n",
      "17 7670723 . A C\n",
      "17 7673738 . C T\n",
      "17 7673804 . C T\n",
      "17 7674173 . C G\n",
      "17 7674907 . G A\n",
      "17 7674958 . A C\n",
      "17 7676003 . C A\n",
      "17 7676204 . A C\n",
      "17 7676559 . G A\n",
      "17 7676580 . C T\n",
      "17 7670683 . T C\n",
      "17 7673617 . G A\n",
      "17 7673798 . A C\n",
      "17 7673843 . C G\n",
      "17 7674851 . A C\n",
      "17 7675147 . G A\n",
      "17 7675159 . G A\n",
      "17 7675231 . G T\n",
      "17 7675244 . A C\n",
      "17 7676003 . C T\n",
      "17 7676021 . A G\n",
      "17 7676108 . T A\n",
      "17 7676240 . C T\n",
      "17 7676373 . T C\n",
      "17 7676583 . C G\n",
      "17 7669703 . G C\n",
      "17 7670633 . G C\n",
      "17 7673580 . G C\n",
      "17 7675081 . G A\n",
      "17 7676132 . A C\n",
      "17 7676143 . C T\n",
      "17 7673754 . A G\n",
      "17 7676549 . G T\n",
      "17 7673809 . C A\n",
      "17 7674390 . G A\n",
      "17 7674973 . T G\n",
      "17 7673533 . A G\n",
      "17 7674811 . C T\n",
      "17 7670625 . T C\n",
      "17 7674857 . A G\n",
      "17 7675237 . C A\n",
      "17 7676963 . G A\n",
      "17 7674291 . C G\n",
      "17 7673030 . C T\n",
      "17 7675397 . T C\n",
      "17 7674089 . A C\n",
      "17 7674109 . G A\n",
      "17 7675361 . A G\n",
      "17 7673183 . G A\n",
      "17 7669739 . G A\n",
      "17 7669911 . C T\n",
      "17 7676324 . C T\n",
      "17 7672958 . T A\n",
      "17 7676734 . C T\n",
      "17 7668134 . G A\n",
      "17 7675238 . T G\n",
      "17 7674276 . A G\n",
      "17 7676625 . G C\n",
      "17 7674215 . G A\n",
      "17 7674192 . C T\n",
      "17 7676172 . A G\n",
      "17 7675337 . T A\n",
      "17 7676034 . C T\n",
      "17 7674811 . C A\n",
      "17 7676203 . C A\n",
      "17 7676186 . A G\n",
      "17 7676601 . G C\n",
      "17 7676081 . A G\n",
      "17 7673719 . G C\n",
      "17 7675196 . T G\n",
      "17 7676106 . G C\n",
      "17 7674243 . A T\n",
      "17 7675123 . G T\n",
      "17 7674274 . G A\n",
      "17 7676222 . A C\n",
      "17 7673539 . A T\n",
      "17 7676194 . C G\n",
      "17 7674212 . T A\n",
      "17 7673755 . G T\n",
      "17 7673739 . T G\n",
      "17 7674182 . T C\n",
      "17 7674859 . C G\n",
      "17 7673797 . A C\n",
      "17 7670646 . C A\n",
      "17 7670607 . A T\n",
      "17 7674882 . C A\n",
      "17 7669688 . T C\n",
      "17 7673728 . C G\n",
      "17 7675211 . A T\n",
      "17 7670629 . C A\n",
      "17 7670619 . C T\n",
      "17 7673719 . G A\n",
      "17 7676184 . T G\n",
      "17 7687376 . C T\n",
      "17 7676543 . T A\n",
      "17 7676222 . A T\n",
      "17 7675977 . C A\n",
      "17 7687376 . C G\n",
      "17 7673547 . A T\n",
      "17 7673833 . T A\n",
      "17 7670604 . C G\n",
      "17 7673532 . T A\n",
      "17 7674175 . C G\n",
      "17 7676187 . T A\n",
      "17 7674188 . C G\n",
      "17 7673816 . G C\n",
      "17 7676245 . C T\n",
      "17 7675194 . T A\n",
      "17 7675252 . G A\n",
      "17 7674899 . G A\n",
      "17 7673810 . A T\n",
      "17 7676188 . C A\n",
      "17 7674984 . A T\n",
      "17 7670688 . A T\n",
      "17 7674281 . C T\n",
      "17 7675168 . A C\n",
      "17 7674971 . C G\n",
      "17 7676199 . T G\n",
      "17 7674846 . T C\n",
      "17 7676399 . G C\n",
      "17 7669679 . G T\n",
      "17 7675239 . G A\n",
      "17 7673539 . A C\n",
      "17 7674853 . C G\n",
      "17 7675072 . C G\n",
      "17 7675072 . C A\n",
      "17 7669611 . A G\n",
      "17 7676004 . A G\n",
      "17 7676046 . C T\n",
      "17 7675234 . G T\n",
      "17 7676136 . G A\n",
      "17 7676215 . G A\n",
      "17 7669682 . T G\n",
      "17 7673844 . A G\n",
      "17 7676585 . G A\n",
      "17 7675163 . G A\n",
      "17 7675199 . G C\n",
      "17 7674189 . T G\n",
      "17 7676010 . T C\n",
      "17 7676032 . A T\n",
      "17 7673535 . C G\n",
      "17 7676149 . C A\n",
      "17 7676369 . C A\n",
      "17 7676047 . C A\n",
      "17 7676076 . G C\n",
      "17 7673602 . G A\n",
      "17 7674909 . C A\n",
      "17 7675172 . A T\n",
      "17 7674965 . G C\n",
      "17 7673784 . C A\n",
      "17 7674866 . G T\n",
      "17 7675068 . A C\n",
      "17 7675127 . A T\n",
      "17 7676212 . A T\n",
      "17 7675196 . T C\n",
      "17 7673743 . C G\n",
      "17 7674237 . G T\n",
      "17 7676001 . G C\n",
      "17 7676389 . T A\n",
      "17 7675143 . C G\n",
      "17 7676020 . C G\n",
      "17 7676077 . G T\n",
      "17 7673529 . T C\n",
      "17 7675146 . G C\n",
      "17 7669641 . T A\n",
      "17 7675211 . A G\n",
      "17 7673607 . T C\n",
      "17 7675246 . G A\n",
      "17 7676422 . A T\n",
      "17 7673574 . T C\n",
      "17 7675046 . T C\n",
      "17 7675096 . A G\n",
      "17 7674270 . G A\n",
      "17 7675063 . T G\n",
      "17 7676108 . T G\n",
      "17 7673592 . G T\n",
      "17 7670629 . C G\n",
      "17 7669612 . G A\n",
      "17 7669705 . A G\n",
      "17 7669707 . C T\n",
      "17 7674216 . C T\n",
      "17 7675138 . G T\n",
      "17 7676291 . A T\n",
      "17 7670595 . C T\n",
      "17 7669618 . G A\n",
      "17 7676418 . C T\n",
      "17 7674163 . T C\n",
      "17 7674198 . G A\n",
      "17 7675247 . A C\n",
      "17 7670732 . G T\n",
      "17 7675039 . C T\n",
      "17 7674305 . A C\n",
      "17 7674308 . C T\n",
      "17 7673521 . C G\n",
      "17 7674985 . G A\n",
      "17 7675087 . G A\n",
      "17 7674302 . C T\n",
      "17 7675231 . G A\n",
      "17 7675093 . C G\n",
      "17 7673681 . T C\n",
      "17 7674961 . A G\n",
      "17 7676504 . C T\n",
      "17 7675982 . A G\n",
      "17 7673621 . C T\n",
      "17 7676419 . A C\n",
      "17 7673601 . G A\n",
      "17 7673688 . C T\n",
      "17 7674989 . G A\n",
      "17 7673705 . C T\n",
      "17 7676562 . C T\n",
      "17 7670721 . G C\n",
      "17 7673522 . C T\n",
      "17 7673807 . C T\n",
      "17 7673621 . C A\n",
      "17 7675165 . G A\n",
      "17 7675041 . C G\n",
      "17 7673849 . A G\n",
      "17 7673776 . G T\n",
      "17 7676597 . C T\n",
      "17 7676341 . T C\n",
      "17 7674255 . G C\n",
      "17 7676240 . C G\n",
      "17 7675197 . T G\n",
      "17 7673750 . G A\n",
      "17 7676498 . G A\n",
      "17 7673593 . G A\n",
      "17 7673853 . A C\n",
      "17 7676475 . G A\n",
      "17 7670578 . G A\n",
      "17 7674176 . C G\n",
      "17 7670642 . C G\n",
      "17 7676256 . T C\n",
      "17 7676172 . A C\n",
      "17 7676140 . G C\n",
      "17 7675229 . G C\n",
      "17 7673744 . T G\n",
      "17 7673599 . T A\n",
      "17 7673564 . G C\n",
      "17 7676230 . G C\n",
      "17 7675206 . G A\n",
      "17 7676046 . C A\n",
      "17 7676013 . G T\n",
      "17 7675996 . T C\n",
      "17 7675225 . G A\n",
      "17 7669627 . T A\n",
      "17 7675203 . G A\n",
      "17 7669646 . T A\n",
      "17 7675228 . A T\n",
      "17 7675226 . G T\n",
      "17 7669629 . C G\n",
      "17 7675174 . C G\n",
      "17 7676054 . G T\n",
      "17 7676038 . G C\n",
      "17 7676558 . G C\n",
      "17 7669636 . G T\n",
      "17 7669636 . G A\n",
      "17 7676042 . G A\n",
      "17 7675241 . A C\n",
      "17 7675236 . A G\n",
      "17 7675146 . G T\n",
      "17 7669609 . T C\n",
      "17 7675093 . C A\n",
      "17 7676599 . G C\n",
      "17 7674930 . A T\n",
      "17 7675152 . C A\n",
      "17 7675128 . T C\n",
      "17 7675102 . C G\n",
      "17 7675100 . T C\n",
      "17 7674938 . T C\n",
      "17 7674934 . T G\n",
      "17 7674909 . C T\n",
      "17 7674906 . T G\n",
      "17 7674878 . A T\n",
      "17 7674877 . C A\n",
      "17 7675156 . C A\n",
      "17 7675109 . T A\n",
      "17 7675108 . G C\n",
      "17 7675084 . G T\n",
      "17 7675057 . G C\n",
      "17 7674920 . T A\n",
      "17 7674888 . T A\n",
      "17 7669614 . C A\n",
      "17 7675111 . C T\n",
      "17 7674871 . A G\n",
      "17 7674240 . G A\n",
      "17 7674236 . T A\n",
      "17 7674209 . G C\n",
      "17 7674207 . G A\n",
      "17 7674202 . A G\n",
      "17 7673828 . T A\n",
      "17 7673821 . G C\n",
      "17 7673789 . A C\n",
      "17 7676391 . G A\n",
      "17 7673701 . C G\n",
      "17 7673599 . T C\n",
      "17 7674215 . G T\n",
      "17 7674215 . G C\n",
      "17 7676521 . A C\n",
      "17 7674212 . T C\n",
      "17 7670689 . C A\n",
      "17 7674177 . T G\n",
      "17 7673840 . A G\n",
      "17 7673840 . A C\n",
      "17 7673836 . C G\n",
      "17 7676397 . A C\n",
      "17 7673792 . G T\n",
      "17 7676378 . T C\n",
      "17 7676242 . A G\n",
      "17 7673559 . C G\n",
      "17 7674254 . T G\n",
      "17 7674252 . C G\n",
      "17 7674225 . C G\n",
      "17 7674219 . C T\n",
      "17 7674218 . T G\n",
      "17 7673770 . T G\n",
      "17 7673759 . C T\n",
      "17 7673723 . C T\n",
      "17 7673723 . C A\n",
      "17 7673721 . G A\n",
      "17 7676384 . G C\n",
      "17 7674334 . G A\n",
      "17 7676523 . T C\n",
      "17 7674195 . T G\n",
      "17 7673819 . C A\n",
      "17 7673732 . G A\n",
      "17 7673593 . G T\n",
      "17 7673598 . G A\n",
      "17 7676277 . A G\n",
      "17 7670714 . A T\n",
      "17 7670714 . A G\n",
      "17 7670685 . G C\n",
      "17 7676212 . A G\n",
      "17 7676211 . C A\n",
      "17 7670650 . G T\n",
      "17 7673508 . T C\n",
      "17 7670670 . C A\n",
      "17 7670663 . T C\n",
      "17 7670653 . A C\n",
      "17 7676197 . G C\n",
      "17 7676189 . T C\n",
      "17 7676238 . A T\n",
      "17 7676238 . A G\n",
      "17 7670659 . G A\n",
      "17 7676199 . T A\n",
      "17 7670679 . G T\n",
      "17 7676194 . C T\n",
      "17 7676192 . A T\n",
      "17 7670646 . C T\n",
      "17 7676173 . T C\n",
      "17 7670631 . C G\n",
      "17 7676261 . C A\n",
      "17 7676141 . T C\n",
      "17 7676132 . A G\n",
      "17 7676130 . G A\n",
      "17 7676572 . G C\n",
      "17 7669694 . T C\n",
      "17 7669690 . G A\n",
      "17 7676118 . G A\n",
      "17 7670615 . T G\n",
      "17 7676123 . C G\n",
      "17 7676111 . T C\n",
      "17 7676109 . G C\n",
      "17 7676148 . G C\n",
      "17 7676126 . T A\n",
      "17 7676113 . C G\n",
      "17 7676099 . G C\n",
      "17 7676576 . C T\n",
      "17 7676595 . G A\n",
      "17 7669682 . T A\n",
      "17 7676259 . G C\n",
      "17 7676088 . G T\n",
      "17 7676088 . G A\n",
      "17 7669670 . C A\n",
      "17 7676079 . A G\n",
      "17 7676078 . G T\n",
      "17 7676065 . T C\n",
      "17 7676063 . G T\n",
      "17 7676085 . G A\n",
      "17 7676081 . A C\n",
      "17 7669666 . C A\n",
      "17 7669668 . G T\n",
      "17 7669691 . C G\n",
      "17 7670732 . G C\n",
      "17 7675260 . A T\n",
      "17 7673691 . G C\n",
      "17 7676283 . G T\n",
      "17 7674955 . C T\n",
      "17 7674965 . G T\n",
      "17 7676060 . G T\n",
      "17 7673518 . G A\n",
      "17 7676151 . A G\n",
      "17 7674237 . G A\n",
      "17 7675095 . C G\n",
      "17 7674839 . A T\n",
      "17 7676501 . C T\n",
      "17 7675248 . G A\n",
      "17 7675137 . C T\n",
      "17 7675162 . T A\n",
      "17 7673731 . G A\n",
      "17 7676142 . G A\n",
      "17 7673537 . G C\n",
      "17 7675201 . C T\n",
      "17 7670626 . C A\n",
      "17 7670628 . C T\n",
      "17 7673526 . A G\n",
      "17 7673527 . C G\n",
      "17 7670629 . C T\n",
      "17 7673562 . T G\n",
      "17 7673541 . G A\n",
      "17 7676146 . G T\n",
      "17 7670613 . A T\n",
      "17 7674982 . T C\n",
      "17 7675127 . A G\n",
      "17 7673852 . A G\n",
      "17 7670643 . C A\n",
      "17 7673774 . C G\n",
      "17 7676365 . C T\n",
      "17 7675975 . G C\n",
      "17 7673747 . C T\n",
      "17 7674174 . T A\n",
      "17 7675978 . C A\n",
      "17 7674933 . T G\n",
      "17 7670733 . G A\n",
      "17 7675218 . T A\n",
      "17 7674285 . G A\n",
      "17 7676254 . C A\n",
      "17 7675189 . G A\n",
      "17 7673702 . T A\n",
      "17 7674886 . A T\n",
      "17 7675986 . C G\n",
      "17 7674971 . C A\n",
      "17 7675081 . G T\n",
      "17 7675983 . G A\n",
      "17 7675130 . G A\n",
      "17 7674228 . G A\n",
      "17 7673685 . T C\n",
      "17 7676554 . A T\n",
      "17 7676365 . C A\n",
      "17 7676122 . C T\n",
      "17 7673724 . A G\n",
      "17 7674979 . C A\n",
      "17 7674306 . G A\n",
      "17 7676055 . C G\n",
      "17 7676385 . A T\n",
      "17 7676231 . G A\n",
      "17 7673842 . T G\n",
      "17 7676507 . A T\n",
      "17 7675181 . T C\n",
      "17 7676239 . T G\n",
      "17 7670623 . G T\n",
      "17 7673575 . G C\n",
      "17 7675997 . G A\n",
      "17 7673618 . A T\n",
      "17 7673856 . G A\n",
      "17 7673562 . T C\n",
      "17 7674243 . A G\n",
      "17 7669617 . A G\n",
      "17 7673516 . A T\n",
      "17 7676570 . T C\n",
      "17 7674854 . C G\n",
      "17 7673813 . G C\n",
      "17 7673604 . C A\n",
      "17 7670730 . G A\n",
      "17 7673515 . G A\n",
      "17 7674290 . C A\n",
      "17 7676005 . C A\n",
      "17 7676030 . G T\n",
      "17 7673757 . T G\n",
      "17 7669652 . T C\n",
      "17 7670652 . C A\n",
      "17 7676205 . G C\n",
      "17 7669621 . A C\n",
      "17 7673830 . G A\n",
      "17 7674201 . G C\n",
      "17 7675214 . A T\n",
      "17 7670003 . C T\n",
      "17 7670658 . T A\n",
      "17 7674400 . C T\n",
      "17 7674179 . A G\n",
      "17 7675099 . C A\n",
      "17 7674933 . T A\n",
      "17 7676584 . G T\n",
      "17 7676019 . C T\n",
      "17 7674904 . T G\n",
      "17 7673590 . C T\n",
      "17 7670660 . A G\n",
      "17 7670703 . C G\n",
      "17 7676036 . C G\n",
      "17 7673556 . A T\n",
      "17 7674281 . C A\n",
      "17 7673610 . T A\n",
      "17 7674179 . A T\n",
      "17 7676078 . G A\n",
      "17 7674858 . C G\n",
      "17 7674244 . C T\n",
      "17 7675063 . T A\n",
      "17 7675118 . T A\n",
      "17 7670612 . G A\n",
      "17 7676075 . A T\n",
      "17 7676377 . C G\n",
      "17 7673814 . C A\n",
      "17 7673576 . G A\n",
      "17 7673574 . T A\n",
      "17 7673819 . C G\n",
      "17 7673815 . T A\n",
      "17 7673491 . G T\n",
      "17 7682425 . C G\n",
      "17 7682463 . C A\n",
      "17 7682348 . T C\n",
      "17 7682350 . T A\n",
      "17 7675945 . C T\n",
      "17 7676316 . C T\n",
      "17 7673551 . T A\n",
      "17 7673550 . T C\n",
      "17 7673558 . C T\n",
      "17 7682469 . C G\n",
      "17 7682470 . C A\n",
      "17 7682238 . G T\n",
      "17 7670686 . G C\n",
      "17 7676153 . G C\n",
      "17 7669632 . T G\n",
      "17 7675083 . G C\n",
      "17 7676380 . A C\n",
      "17 7673714 . C T\n",
      "17 7670607 . A G\n",
      "17 7676044 . A T\n",
      "17 7674261 . G T\n",
      "17 7670680 . C G\n",
      "17 7673563 . G C\n",
      "17 7674914 . A C\n",
      "17 7673769 . G C\n",
      "17 7673594 . T C\n",
      "17 7675167 . A G\n",
      "17 7673847 . A C\n",
      "17 7673561 . G C\n",
      "17 7673744 . T A\n",
      "17 7674957 . G C\n",
      "17 7673727 . T G\n",
      "17 7674187 . T G\n",
      "17 7674255 . G T\n",
      "17 7670669 . G C\n",
      "17 7675245 . A G\n",
      "17 7675042 . C A\n",
      "17 7675171 . A C\n",
      "17 7675222 . G C\n",
      "17 7675159 . G T\n",
      "17 7675110 . G T\n",
      "17 7675230 . G A\n",
      "17 7675214 . A C\n",
      "17 7675155 . G C\n",
      "17 7675062 . C G\n",
      "17 7675185 . C G\n",
      "17 7675184 . A T\n",
      "17 7675169 . T A\n",
      "17 7675158 . G T\n",
      "17 7675122 . T G\n",
      "17 7675116 . A C\n",
      "17 7675018 . C T\n",
      "17 7675242 . G C\n",
      "17 7675247 . A G\n",
      "17 7675240 . T G\n",
      "17 7675284 . A T\n",
      "17 7676199 . T C\n",
      "17 7676030 . G C\n",
      "17 7674959 . G A\n",
      "17 7676206 . T G\n",
      "17 7676532 . G C\n",
      "17 7673686 . G A\n",
      "17 7674964 . G C\n",
      "17 7676413 . A G\n",
      "17 7670641 . C G\n",
      "17 7687376 . C A\n",
      "17 7673520 . C G\n",
      "17 7669670 . C T\n",
      "17 7673692 . C G\n",
      "17 7676123 . C A\n",
      "17 7676414 . A G\n",
      "17 7676061 . T G\n",
      "17 7675113 . G T\n",
      "17 7670665 . C T\n",
      "17 7674982 . T G\n",
      "17 7674845 . G A\n",
      "17 7675183 . C T\n",
      "17 7673834 . A G\n",
      "17 7676193 . C G\n",
      "17 7674285 . G C\n",
      "17 7674937 . T C\n",
      "17 7670678 . A C\n",
      "17 7674876 . G T\n",
      "17 7675243 . G T\n",
      "17 7670666 . A G\n",
      "17 7670632 . T G\n",
      "17 7673821 . G T\n",
      "17 7675124 . T G\n",
      "17 7669700 . A G\n",
      "17 7675166 . G T\n",
      "17 7674908 . T G\n",
      "17 7674247 . T A\n",
      "17 7673791 . A C\n",
      "17 7674184 . G A\n",
      "17 7673623 . A G\n",
      "17 7669696 . G C\n",
      "17 7670612 . G T\n",
      "17 7676371 . C T\n",
      "17 7673826 . A C\n",
      "17 7673767 . C A\n",
      "17 7675149 . T G\n",
      "17 7673620 . G A\n",
      "17 7670605 . T A\n",
      "17 7670643 . C T\n",
      "17 7670645 . G T\n",
      "17 7673615 . A C\n",
      "17 7674254 . T A\n",
      "17 7675068 . A G\n",
      "17 7675179 . G C\n",
      "17 7676179 . G A\n",
      "17 7676242 . A C\n",
      "17 7673548 . T C\n",
      "17 7675186 . A G\n",
      "17 7675079 . T C\n",
      "17 7670733 . G T\n",
      "17 7676226 . T C\n",
      "17 7674190 . T A\n",
      "17 7673701 . C T\n",
      "17 7675159 . G C\n",
      "17 7676566 . A T\n",
      "17 7676115 . G A\n",
      "17 7674172 . G T\n",
      "17 7676194 . C A\n",
      "17 7675072 . C T\n",
      "17 7674310 . C A\n",
      "17 7676029 . A G\n",
      "17 7676091 . A C\n",
      "17 7674196 . G T\n",
      "17 7670639 . T G\n",
      "17 7676369 . C G\n",
      "17 7673541 . G T\n",
      "17 7674269 . T A\n",
      "17 7676423 . A G\n",
      "17 7673711 . G A\n",
      "17 7675036 . C T\n",
      "17 7676287 . A C\n",
      "17 7676110 . G A\n",
      "17 7676116 . G C\n",
      "17 7673520 . C T\n",
      "17 7673535 . C A\n",
      "17 7670728 . A C\n",
      "17 7670723 . A T\n",
      "17 7673565 . T C\n",
      "17 7676406 . G A\n",
      "17 7675151 . C G\n",
      "17 7673817 . T C\n",
      "17 7673600 . T G\n",
      "17 7675129 . G A\n",
      "17 7670731 . A C\n",
      "17 7674987 . G A\n",
      "17 7670655 . C G\n",
      "17 7676585 . G C\n",
      "17 7669710 . G A\n",
      "17 7675120 . C A\n",
      "17 7674876 . G C\n",
      "17 7676110 . G C\n",
      "17 7673624 . G A\n",
      "17 7673688 . C A\n",
      "17 7676063 . G C\n",
      "17 7673808 . T C\n",
      "17 7675976 . C A\n",
      "17 7673614 . A G\n",
      "17 7673726 . C G\n",
      "17 7669633 . C T\n",
      "17 7676221 . T G\n",
      "17 7675107 . T A\n",
      "17 7673585 . A G\n",
      "17 7674273 . G A\n",
      "17 7670589 . A G\n",
      "17 7675133 . A G\n",
      "17 7674204 . G A\n",
      "17 7675034 . T A\n",
      "17 7674912 . C T\n",
      "17 7674171 . G T\n",
      "17 7676117 . G T\n",
      "17 7673835 . C T\n",
      "17 7676370 . C T\n",
      "17 7670672 . T C\n",
      "17 7674335 . G A\n",
      "17 7668194 . C T\n",
      "17 7668195 . G A\n",
      "17 7668440 . T C\n",
      "17 7669658 . G T\n",
      "17 7675251 . A G\n",
      "17 7676177 . G A\n",
      "17 7676266 . A G\n",
      "17 7676170 . G C\n",
      "17 7673614 . A C\n",
      "17 7670720 . A C\n",
      "17 7670686 . G T\n",
      "17 7673569 . T G\n",
      "17 7676277 . A C\n",
      "17 7673814 . C T\n",
      "17 7673808 . T A\n",
      "17 7676257 . G A\n",
      "17 7674973 . T C\n",
      "17 7674180 . C G\n",
      "17 7676274 . T G\n",
      "17 7674972 . C A\n",
      "17 7674914 . A T\n",
      "17 7670608 . C A\n",
      "17 7673610 . T G\n",
      "17 7674206 . T G\n",
      "17 7670648 . T A\n",
      "17 7669643 . A G\n",
      "17 7669620 . C T\n",
      "17 7669610 . C A\n",
      "17 7676203 . C T\n",
      "17 7676097 . C G\n",
      "17 7675177 . C T\n",
      "17 7674902 . T C\n",
      "17 7674249 . A T\n",
      "17 7673735 . A T\n",
      "17 7673503 . T A\n",
      "17 7670655 . C T\n",
      "17 7670689 . C G\n",
      "17 7673678 . C T\n",
      "17 7674835 . A G\n",
      "17 7675065 . A G\n",
      "17 7675109 . T G\n",
      "17 7670705 . C A\n",
      "17 7673715 . C G\n",
      "17 7675166 . G C\n",
      "17 7669672 . C G\n",
      "17 7673572 . T G\n",
      "17 7670648 . T G\n",
      "17 7674970 . A C\n",
      "17 7673704 . G C\n",
      "17 7673736 . G T\n",
      "17 7673746 . T A\n",
      "17 7674319 . G C\n",
      "17 7676091 . A G\n",
      "17 7675244 . A G\n",
      "17 7674941 . A G\n",
      "17 7676239 . T C\n",
      "17 7676261 . C G\n",
      "17 7674919 . C G\n",
      "17 7676219 . A G\n",
      "17 7676089 . A C\n",
      "17 7676000 . A T\n",
      "17 7676234 . C G\n",
      "17 7676151 . A T\n",
      "17 7674925 . A T\n",
      "17 7676016 . G C\n",
      "17 7675192 . G C\n",
      "17 7675104 . T A\n",
      "17 7673800 . C T\n",
      "17 7676195 . T A\n",
      "17 7670680 . C A\n",
      "17 7670608 . C T\n",
      "17 7676142 . G C\n",
      "17 7670653 . A G\n",
      "17 7670642 . C T\n",
      "17 7674896 . A G\n",
      "17 7676153 . G T\n",
      "17 7673708 . A C\n",
      "17 7675068 . A T\n",
      "17 7669605 . A C\n",
      "17 7675172 . A C\n",
      "17 7670703 . C A\n",
      "17 7676552 . T C\n",
      "17 7673524 . A G\n",
      "17 7675228 . A C\n",
      "17 7673702 . T C\n",
      "17 7669613 . T C\n",
      "17 7674165 . G A\n",
      "17 7669613 . T A\n",
      "17 7669662 . T A\n",
      "17 7673603 . G T\n",
      "17 7673709 . G C\n",
      "17 7673722 . G A\n",
      "17 7673809 . C G\n",
      "17 7674188 . C T\n",
      "17 7674283 . G A\n",
      "17 7675119 . G C\n",
      "17 7676032 . A G\n",
      "17 7676034 . C G\n",
      "17 7676089 . A T\n",
      "17 7676094 . G T\n",
      "17 7676152 . C A\n",
      "17 7676178 . G A\n",
      "17 7676390 . T C\n",
      "17 7676396 . C T\n",
      "17 7676534 . C T\n",
      "17 7670635 . C T\n",
      "17 7674162 . G T\n",
      "17 7676018 . C G\n",
      "17 7674300 . A G\n",
      "17 7676159 . A C\n",
      "17 7675254 . A C\n",
      "17 7676574 . A G\n",
      "17 7675207 . G T\n",
      "17 7676291 . A G\n",
      "17 7676279 . A G\n",
      "17 7670733 . G C\n",
      "17 7675115 . G T\n",
      "17 7676012 . G C\n",
      "17 7670718 . G C\n",
      "17 7674293 . A C\n",
      "17 7670727 . G C\n",
      "17 7676526 . C T\n",
      "17 7674259 . T A\n",
      "17 7673687 . T A\n",
      "17 7676506 . A G\n",
      "17 7670675 . T C\n",
      "17 7675986 . C T\n",
      "17 7674841 . C T\n",
      "17 7674985 . G C\n",
      "17 7676147 . G C\n",
      "17 7675084 . G A\n",
      "17 7676506 . A C\n",
      "17 7676179 . G T\n",
      "17 7674182 . T G\n",
      "17 7673517 . G C\n",
      "17 7670722 . C T\n",
      "17 7673856 . G C\n",
      "17 7673519 . T C\n",
      "17 7675041 . C A\n",
      "17 7676371 . C G\n",
      "17 7673628 . A T\n",
      "17 7676292 . G C\n",
      "17 7669716 . G A\n",
      "17 7669733 . G A\n",
      "17 7673669 . C T\n",
      "17 7674319 . G A\n",
      "17 7674323 . G A\n",
      "17 7674825 . C T\n",
      "17 7675021 . G T\n",
      "17 7675968 . G T\n",
      "17 7676453 . A G\n",
      "17 7676482 . T G\n",
      "17 7676620 . C T\n",
      "17 7670702 . T C\n",
      "17 7676010 . T G\n",
      "17 7674846 . T G\n",
      "17 7674329 . C T\n",
      "17 7669601 . G A\n",
      "17 7674943 . T C\n",
      "17 7674279 . G A\n",
      "17 7675066 . G T\n",
      "17 7675116 . A T\n",
      "17 7675997 . G C\n",
      "17 7676184 . T C\n",
      "17 7674910 . A C\n",
      "17 7675188 . G C\n",
      "17 7673698 . T A\n",
      "17 7676561 . G A\n",
      "17 7673793 . G A\n",
      "17 7673566 . T G\n",
      "17 7675208 . C G\n",
      "17 7673762 . T G\n",
      "17 7676202 . T C\n",
      "17 7676122 . C A\n",
      "17 7673838 . C G\n",
      "17 7666193 . C T\n",
      "17 7675205 . T A\n",
      "17 7674966 . C T\n",
      "17 7675188 . G T\n",
      "17 7676122 . C G\n",
      "17 7669659 . A T\n",
      "17 7674907 . G C\n",
      "17 7674913 . C G\n",
      "17 7675192 . G A\n",
      "17 7676533 . T G\n",
      "17 7674854 . C A\n",
      "17 7673550 . T G\n",
      "17 7675202 . A T\n",
      "17 7674245 . T A\n",
      "17 7676117 . G A\n",
      "17 7674911 . T C\n",
      "17 7675999 . A G\n",
      "17 7676090 . C G\n",
      "17 7676421 . A G\n",
      "17 7673627 . A G\n",
      "17 7673607 . T G\n",
      "17 7676216 . T C\n",
      "17 7674988 . A G\n",
      "17 7676375 . T C\n",
      "17 7673780 . T C\n",
      "17 7669621 . A G\n",
      "17 7676292 . G A\n",
      "17 7674946 . G A\n",
      "17 7675153 . G T\n",
      "17 7674186 . G A\n",
      "17 7675043 . A G\n",
      "17 7674981 . A G\n",
      "17 7676286 . A G\n",
      "17 7670662 . T C\n",
      "17 7676512 . T C\n",
      "17 7676270 . G A\n",
      "17 7673529 . T A\n",
      "17 7676374 . G A\n",
      "17 7676529 . T A\n",
      "17 7669608 . G A\n",
      "17 7675126 . G A\n",
      "17 7676012 . G A\n",
      "17 7670734 . G A\n",
      "17 7676054 . G A\n",
      "17 7673798 . A G\n",
      "17 7676027 . C T\n",
      "17 7669624 . C T\n",
      "17 7675150 . G C\n",
      "17 7674304 . C T\n",
      "17 7674302 . C A\n",
      "17 7676105 . G T\n",
      "17 7676129 . A C\n",
      "17 7674892 . T G\n",
      "17 7676419 . A G\n",
      "17 7676513 . C G\n",
      "17 7676228 . C G\n",
      "17 7676513 . C A\n",
      "17 7673695 . G A\n",
      "17 7676180 . A C\n",
      "17 7674986 . T C\n",
      "17 7675123 . G A\n",
      "17 7674978 . G A\n",
      "17 7675141 . G C\n",
      "17 7675171 . A G\n",
      "17 7674172 . G C\n",
      "17 7676285 . G A\n",
      "17 7674983 . C A\n",
      "17 7675177 . C A\n",
      "17 7676060 . G A\n",
      "17 7670719 . C T\n",
      "17 7675985 . G T\n",
      "17 7670723 . A G\n",
      "17 7676120 . C G\n",
      "17 7676514 . C T\n",
      "17 7673589 . G A\n",
      "17 7669678 . G T\n",
      "17 7676111 . T G\n",
      "17 7674294 . G A\n",
      "17 7676045 . A G\n",
      "17 7674852 . A C\n",
      "17 7673708 . A T\n",
      "17 7669663 . A G\n",
      "17 7670601 . T A\n",
      "17 7675195 . C T\n",
      "17 7674898 . A G\n",
      "17 7676126 . T G\n",
      "17 7674874 . G A\n",
      "17 7669763 . C G\n",
      "17 7679754 . A G\n",
      "17 7673202 . C A\n",
      "17 7670660 . A C\n",
      "17 7676382 . C G\n",
      "17 7675052 . C G\n",
      "17 7674246 . G T\n",
      "17 7676086 . A T\n",
      "17 7669689 . G C\n",
      "17 7670715 . T G\n",
      "17 7675089 . G T\n",
      "17 7674968 . A G\n",
      "17 7676133 . G A\n",
      "17 7676070 . T G\n",
      "17 7673762 . T C\n",
      "17 7669609 . T G\n",
      "17 7675221 . T G\n",
      "17 7677724 . A G\n"
     ]
    }
   ],
   "source": [
    "chunk = 200\n",
    "for i in variantsParams:\n",
    "    print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "ac340bff",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Connection status: 200\n",
      "[{'assembly_name': 'GRCh38', 'seq_region_name': '17', 'strand': 1, 'end': 7674221, 'input': '17 7674221 . G A', 'colocated_variants': [{'strand': 1, 'seq_region_name': '17', 'id': 'CM010465', 'start': 7674221, 'clinvar_somatic_classification': 'Neoplasm(oncogenicity:Oncogenic)', 'allele_string': 'HGMD_MUTATION', 'phenotype_or_disease': 1, 'end': 7674221}, {'seq_region_name': '17', 'id': 'CM900211', 'strand': 1, 'end': 7674221, 'phenotype_or_disease': 1, 'allele_string': 'HGMD_MUTATION', 'clinvar_somatic_classification': 'Neoplasm(oncogenicity:Oncogenic)', 'start': 7674221}, {'seq_region_name': '17', 'var_synonyms': {'COSMIC': ['COSM10656']}, 'id': 'COSV52662035', 'strand': 1, 'end': 7674221, 'somatic': 1, 'allele_string': 'COSMIC_MUTATION', 'phenotype_or_disease': 1, 'start': 7674221, 'clinvar_somatic_classification': 'Neoplasm(oncogenicity:Oncogenic)'}, {'somatic': 1, 'end': 7674221, 'phenotype_or_disease': 1, 'allele_string': 'COSMIC_MUTATION', 'start': 7674221, 'clinvar_somatic_classification': 'Neoplasm(oncogenicity:Oncogenic)', 'id': 'COSV52797251', 'var_synonyms': {'COSMIC': ['COSM11564']}, 'seq_region_name': '17', 'strand': 1}, {'strand': 1, 'seq_region_name': '17', 'var_synonyms': {'COSMIC': ['COSM44920']}, 'id': 'COSV53097881', 'phenotype_or_disease': 1, 'allele_string': 'COSMIC_MUTATION', 'clinvar_somatic_classification': 'Neoplasm(oncogenicity:Oncogenic)', 'start': 7674221, 'end': 7674221, 'somatic': 1}, {'end': 7674221, 'frequencies': {'A': {'gnomadg_afr': 0, 'gnomade': 6.157e-06, 'gnomadg': 0, 'gnomade_amr': 2.236e-05, 'gnomadg_asj': 0, 'gnomadg_ami': 0, 'gnomadg_sas': 0, 'gnomade_mid': 0, 'gnomade_asj': 0, 'gnomadg_nfe': 0, 'gnomade_afr': 0, 'gnomadg_fin': 0, 'gnomade_sas': 1.159e-05, 'gnomadg_eas': 0, 'gnomadg_remaining': 0, 'gnomadg_amr': 0, 'gnomade_remaining': 0, 'gnomade_fin': 0, 'gnomade_eas': 0, 'gnomadg_mid': 0, 'gnomade_nfe': 6.296e-06}}, 'clinvar_somatic_classification': 'Neoplasm(oncogenicity:Oncogenic)', 'start': 7674221, 'allele_string': 'G/A/C/T', 'phenotype_or_disease': 1, 'clin_sig_allele': 'T:likely_benign;A:pathogenic/likely_pathogenic;C:pathogenic;A:uncertain_significance;C:likely_pathogenic;A:pathogenic', 'var_synonyms': {'OMIM': [191170.0001], 'PhenCode': ['TP53_g.13379C>T'], 'ClinVar': ['RCV000444845', 'RCV000442243', 'RCV000441557', 'RCV000441091', 'RCV000440560', 'RCV000440422', 'RCV000438698', 'RCV000436398', 'RCV000435803', 'RCV000434504', 'RCV000433905', 'RCV000431689', 'RCV000431508', 'RCV000430543', 'RCV000429884', 'RCV000429777', 'RCV000425682', 'RCV000425083', 'RCV000424415', 'RCV000424308', 'RCV000423804', 'RCV000423184', 'RCV000419857', 'RCV000419150', 'RCV000419032', 'RCV000418495', 'RCV000213057', 'RCV000168242', 'RCV000115735', 'RCV000013140', 'VCV000012347', 'RCV000785485', 'RCV000735293', 'RCV001374442', 'RCV001271100', 'RCV001255674', 'RCV003162243', 'RCV003149569', 'RCV003105772', 'RCV000626118', 'RCV000444519', 'RCV000444427', 'RCV000444356', 'RCV000441711', 'RCV000441010', 'RCV000440334', 'RCV000437882', 'RCV000436038', 'RCV000435353', 'RCV000435050', 'RCV000433611', 'RCV000432931', 'RCV000432207', 'RCV000633396', 'RCV000430735', 'RCV000427948', 'RCV000427544', 'RCV000425782', 'RCV000425394', 'RCV000425100', 'RCV000424776', 'RCV000423297', 'RCV000422668', 'RCV000420498', 'RCV000418362', 'RCV000417693', 'VCV000376652', 'RCV001026425', 'RCV004022238', 'RCV001175916', 'VCV000918353', 'RCV003460463', 'RCV003318332', 'RCV004745154'], 'UniPro': []}, 'id': 'rs121912651', 'seq_region_name': '17', 'pubmed': [25741868, 24033266, 26900293, 32293552, 25157968, 26619011, 25105660, 16489069, 17606709, 30744407, 1631137, 1978757, 21343334, 8099841, 8425176, 8527048, 9598730, 9825943, 12826609, 17427234, 20013323, 28664506, 37182128, 34771529, 34178683, 36866106, 35483882, 37391803, 38310289, 38413718, 39457600, 39208653, 39488539], 'clin_sig': ['uncertain_significance', 'likely_benign', 'likely_pathogenic', 'pathogenic'], 'strand': 1}], 'allele_string': 'G/A', 'transcript_consequences': [{'strand': -1, 'cdna_end': 884, 'gene_symbol': 'TP53', 'sift_score': 0, 'cdna_start': 884, 'sift_prediction': 'deleterious', 'cds_start': 742, 'canonical': 1, 'biotype': 'protein_coding', 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'protein_start': 248, 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'variant_allele': 'A', 'cds_end': 742, 'cadd_raw': 5.203168, 'impact': 'MODERATE', 'blosum62': -3, 'transcript_id': 'ENST00000269305', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'cadd_phred': 29.2, 'revel': 0.927, 'codons': 'Cgg/Tgg', 'protein_end': 248}, {'cadd_raw': 5.203168, 'blosum62': -3, 'impact': 'MODERATE', 'transcript_id': 'ENST00000359597', 'gene_symbol_source': 'HGNC', 'polyphen_score': 1, 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'revel': 0.927, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'protein_end': 248, 'strand': -1, 'cdna_end': 742, 'gene_symbol': 'TP53', 'sift_score': 0, 'cdna_start': 742, 'cds_start': 742, 'polyphen_prediction': 'probably_damaging', 'sift_prediction': 'deleterious', 'biotype': 'protein_coding', 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 248, 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'variant_allele': 'A', 'cds_end': 742}, {'amino_acids': 'R/W', 'polyphen_score': 1, 'gene_id': 'ENSG00000141510', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000413465', 'cadd_raw': 5.203168, 'blosum62': -3, 'impact': 'MODERATE', 'protein_end': 248, 'codons': 'Cgg/Tgg', 'revel': 0.927, 'cadd_phred': 29.2, 'polyphen_prediction': 'probably_damaging', 'sift_prediction': 'deleterious', 'cds_start': 742, 'cdna_start': 742, 'sift_score': 0, 'gene_symbol': 'TP53', 'cdna_end': 742, 'strand': -1, 'cds_end': 742, 'variant_allele': 'A', 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'protein_start': 248, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'biotype': 'protein_coding'}, {'transcript_id': 'ENST00000420246', 'blosum62': -3, 'cadd_raw': 5.203168, 'impact': 'MODERATE', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'gene_symbol_source': 'HGNC', 'revel': 0.927, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'protein_end': 248, 'cdna_end': 875, 'strand': -1, 'sift_prediction': 'deleterious', 'cds_start': 742, 'gene_symbol': 'TP53', 'sift_score': 0, 'cdna_start': 875, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'protein_start': 248, 'biotype': 'protein_coding', 'cds_end': 742, 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'variant_allele': 'A'}, {'revel': 0.927, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'protein_end': 248, 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000445888', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'biotype': 'protein_coding', 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 248, 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'variant_allele': 'A', 'cds_end': 742, 'strand': -1, 'cdna_end': 878, 'gene_symbol': 'TP53', 'cdna_start': 878, 'sift_score': 0, 'sift_prediction': 'deleterious', 'cds_start': 742}, {'gene_symbol': 'TP53', 'cdna_start': 875, 'sift_score': 0, 'sift_prediction': 'deleterious', 'cds_start': 742, 'strand': -1, 'cdna_end': 875, 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'variant_allele': 'A', 'cds_end': 742, 'biotype': 'protein_coding', 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 248, 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000455263', 'protein_end': 248, 'cadd_phred': 29.2, 'revel': 0.927, 'codons': 'Cgg/Tgg'}, {'cdna_end': 924, 'strand': -1, 'sift_prediction': 'deleterious', 'cds_start': 742, 'gene_symbol': 'TP53', 'cdna_start': 924, 'sift_score': 0, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'protein_start': 248, 'biotype': 'protein_coding', 'cds_end': 742, 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'variant_allele': 'A', 'transcript_id': 'ENST00000503591', 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'blosum62': -3, 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'gene_symbol_source': 'HGNC', 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'protein_end': 248}, {'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 116, 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000504290', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'biotype': 'protein_coding', 'protein_start': 116, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'variant_allele': 'A', 'cds_end': 346, 'strand': -1, 'cdna_end': 624, 'cdna_start': 624, 'sift_score': 0, 'gene_symbol': 'TP53', 'cds_start': 346, 'sift_prediction': 'deleterious'}, {'cds_end': 346, 'variant_allele': 'A', 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'protein_start': 116, 'biotype': 'protein_coding', 'cds_start': 346, 'sift_prediction': 'deleterious', 'gene_symbol': 'TP53', 'cdna_start': 624, 'sift_score': 0, 'cdna_end': 624, 'strand': -1, 'protein_end': 116, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000504937', 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'blosum62': -3}, {'cadd_phred': 29.2, 'biotype': 'retained_intron', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:11998', 'variant_allele': 'A', 'impact': 'MODIFIER', 'cadd_raw': 5.203168, 'strand': -1, 'transcript_id': 'ENST00000505014', 'gene_symbol': 'TP53', 'distance': 305, 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510'}, {'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 248, 'biotype': 'protein_coding', 'cds_end': 742, 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'variant_allele': 'A', 'cdna_end': 1024, 'strand': -1, 'cds_start': 742, 'sift_prediction': 'deleterious', 'gene_symbol': 'TP53', 'cdna_start': 1024, 'sift_score': 0, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'protein_end': 248, 'transcript_id': 'ENST00000508793', 'impact': 'MODERATE', 'blosum62': -3, 'cadd_raw': 5.203168, 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'gene_symbol_source': 'HGNC'}, {'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000509690', 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'blosum62': -3, 'protein_end': 116, 'revel': 0.927, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'sift_prediction': 'deleterious', 'cds_start': 346, 'gene_symbol': 'TP53', 'sift_score': 0, 'cdna_start': 478, 'cdna_end': 478, 'strand': -1, 'cds_end': 346, 'variant_allele': 'A', 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 116, 'biotype': 'protein_coding'}, {'biotype': 'protein_coding', 'protein_start': 116, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'variant_allele': 'A', 'cds_end': 346, 'strand': -1, 'cdna_end': 624, 'cdna_start': 624, 'sift_score': 0, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 346, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 116, 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000510385', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510'}, {'cdna_end': 542, 'strand': -1, 'sift_prediction': 'deleterious', 'polyphen_prediction': 'probably_damaging', 'cds_start': 463, 'cdna_start': 542, 'sift_score': 0, 'gene_symbol': 'TP53', 'protein_start': 155, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'biotype': 'protein_coding', 'cds_end': 463, 'variant_allele': 'A', 'transcript_id': 'ENST00000514944', 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'blosum62': -3, 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'polyphen_score': 1, 'gene_symbol_source': 'HGNC', 'codons': 'Cgg/Tgg', 'revel': 0.927, 'cadd_phred': 29.2, 'protein_end': 155}, {'variant_allele': 'A', 'biotype': 'protein_coding_CDS_not_defined', 'cadd_phred': 29.2, 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:11998', 'distance': 33, 'gene_symbol_source': 'HGNC', 'gene_symbol': 'TP53', 'gene_id': 'ENSG00000141510', 'impact': 'MODIFIER', 'cadd_raw': 5.203168, 'strand': -1, 'transcript_id': 'ENST00000574684'}, {'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'impact': 'MODERATE', 'blosum62': -3, 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000576024', 'protein_end': 248, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'gene_symbol': 'TP53', 'sift_score': 0, 'cdna_start': 905, 'sift_prediction': 'deleterious', 'cds_start': 742, 'strand': -1, 'cdna_end': 905, 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'variant_allele': 'A', 'cds_end': 742, 'biotype': 'protein_coding', 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 248}, {'variant_allele': 'A', 'cds_end': 721, 'biotype': 'protein_coding', 'protein_start': 241, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'cdna_start': 860, 'sift_score': 0, 'gene_symbol': 'TP53', 'polyphen_prediction': 'probably_damaging', 'sift_prediction': 'deleterious', 'cds_start': 721, 'strand': -1, 'cdna_end': 860, 'protein_end': 241, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'polyphen_score': 0.999, 'cadd_raw': 5.203168, 'blosum62': -3, 'impact': 'MODERATE', 'transcript_id': 'ENST00000604348'}, {'protein_start': 209, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'biotype': 'protein_coding', 'cds_end': 625, 'variant_allele': 'A', 'cdna_end': 992, 'strand': -1, 'sift_prediction': 'deleterious', 'cds_start': 625, 'sift_score': 0, 'cdna_start': 992, 'gene_symbol': 'TP53', 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 209, 'transcript_id': 'ENST00000610292', 'cadd_raw': 5.203168, 'blosum62': -3, 'impact': 'MODERATE', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'gene_symbol_source': 'HGNC'}, {'transcript_id': 'ENST00000610538', 'cadd_raw': 5.203168, 'impact': 'MODERATE', 'blosum62': -3, 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'gene_symbol_source': 'HGNC', 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 209, 'cdna_end': 875, 'strand': -1, 'sift_prediction': 'deleterious', 'cds_start': 625, 'sift_score': 0, 'cdna_start': 875, 'gene_symbol': 'TP53', 'protein_start': 209, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'biotype': 'protein_coding', 'cds_end': 625, 'variant_allele': 'A'}, {'cds_end': 265, 'variant_allele': 'A', 'protein_start': 89, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'biotype': 'protein_coding', 'cds_start': 265, 'sift_prediction': 'deleterious', 'cdna_start': 624, 'sift_score': 0, 'gene_symbol': 'TP53', 'cdna_end': 624, 'strand': -1, 'protein_end': 89, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000610623', 'blosum62': -3, 'cadd_raw': 5.203168, 'impact': 'MODERATE'}, {'variant_allele': 'A', 'cds_end': 265, 'biotype': 'protein_coding', 'protein_start': 89, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'sift_score': 0, 'cdna_start': 624, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 265, 'strand': -1, 'cdna_end': 624, 'protein_end': 89, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'cadd_raw': 5.203168, 'blosum62': -3, 'impact': 'MODERATE', 'transcript_id': 'ENST00000618944'}, {'sift_prediction': 'deleterious', 'cds_start': 265, 'gene_symbol': 'TP53', 'cdna_start': 624, 'sift_score': 0, 'cdna_end': 624, 'strand': -1, 'cds_end': 265, 'variant_allele': 'A', 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'protein_start': 89, 'biotype': 'protein_coding', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000619186', 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'protein_end': 89, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg'}, {'cds_end': 625, 'variant_allele': 'A', 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 209, 'biotype': 'protein_coding', 'sift_prediction': 'deleterious', 'cds_start': 625, 'gene_symbol': 'TP53', 'cdna_start': 878, 'sift_score': 0, 'cdna_end': 878, 'strand': -1, 'protein_end': 209, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000619485', 'cadd_raw': 5.203168, 'impact': 'MODERATE', 'blosum62': -3}, {'cadd_raw': 5.203168, 'impact': 'MODERATE', 'blosum62': -3, 'transcript_id': 'ENST00000620739', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'protein_end': 209, 'strand': -1, 'cdna_end': 932, 'gene_symbol': 'TP53', 'cdna_start': 932, 'sift_score': 0, 'sift_prediction': 'deleterious', 'cds_start': 625, 'biotype': 'protein_coding', 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'protein_start': 209, 'variant_allele': 'A', 'cds_end': 625}, {'protein_end': 209, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'blosum62': -3, 'cadd_raw': 5.203168, 'impact': 'MODERATE', 'transcript_id': 'ENST00000622645', 'variant_allele': 'A', 'cds_end': 625, 'biotype': 'protein_coding', 'protein_start': 209, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'sift_score': 0, 'cdna_start': 875, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 625, 'strand': -1, 'cdna_end': 875}, {'cds_end': 625, 'variant_allele': 'A', 'protein_start': 209, 'consequence_terms': ['missense_variant', 'NMD_transcript_variant'], 'hgnc_id': 'HGNC:11998', 'biotype': 'nonsense_mediated_decay', 'polyphen_prediction': 'probably_damaging', 'sift_prediction': 'deleterious_low_confidence', 'cds_start': 625, 'cdna_start': 885, 'sift_score': 0, 'gene_symbol': 'TP53', 'cdna_end': 885, 'strand': -1, 'protein_end': 209, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'amino_acids': 'R/W', 'polyphen_score': 1, 'gene_id': 'ENSG00000141510', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000635293', 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168}, {'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 209, 'cadd_raw': 5.203168, 'blosum62': -3, 'impact': 'MODERATE', 'transcript_id': 'ENST00000714356', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'biotype': 'protein_coding', 'protein_start': 209, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'variant_allele': 'A', 'cds_end': 625, 'strand': -1, 'cdna_end': 1005, 'sift_score': 0, 'cdna_start': 1005, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 625}, {'cds_end': 742, 'variant_allele': 'A', 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'protein_start': 248, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'biotype': 'protein_coding', 'cds_start': 742, 'sift_prediction': 'deleterious', 'cdna_start': 812, 'sift_score': 0, 'gene_symbol': 'TP53', 'cdna_end': 812, 'strand': -1, 'protein_end': 248, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000714357', 'cadd_raw': 5.203168, 'blosum62': -3, 'impact': 'MODERATE'}, {'variant_allele': 'A', 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['3_prime_UTR_variant', 'NMD_transcript_variant'], 'biotype': 'nonsense_mediated_decay', 'cadd_phred': 29.2, 'gene_id': 'ENSG00000141510', 'gene_symbol_source': 'HGNC', 'cdna_start': 862, 'gene_symbol': 'TP53', 'cdna_end': 862, 'transcript_id': 'ENST00000714358', 'cadd_raw': 5.203168, 'impact': 'MODIFIER', 'strand': -1}, {'cdna_start': 859, 'sift_score': 0, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 742, 'strand': -1, 'cdna_end': 859, 'variant_allele': 'A', 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'cds_end': 742, 'biotype': 'protein_coding', 'protein_start': 248, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'blosum62': -3, 'transcript_id': 'ENST00000714359', 'protein_end': 248, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2}, {'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'protein_end': 248, 'impact': 'MODERATE', 'blosum62': -3, 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000714408', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'biotype': 'protein_coding', 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 248, 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'variant_allele': 'A', 'cds_end': 742, 'strand': -1, 'cdna_end': 884, 'gene_symbol': 'TP53', 'cdna_start': 884, 'sift_score': 0, 'cds_start': 742, 'sift_prediction': 'deleterious'}, {'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'variant_allele': 'A', 'cds_end': 742, 'biotype': 'protein_coding', 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'protein_start': 248, 'gene_symbol': 'TP53', 'cdna_start': 884, 'sift_score': 0, 'sift_prediction': 'deleterious', 'cds_start': 742, 'strand': -1, 'cdna_end': 884, 'protein_end': 248, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'blosum62': -3, 'cadd_raw': 5.203168, 'impact': 'MODERATE', 'transcript_id': 'ENST00000714409'}, {'biotype': 'protein_coding', 'protein_start': 248, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'variant_allele': 'A', 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'cds_end': 742, 'strand': -1, 'cdna_end': 837, 'cdna_start': 837, 'sift_score': 0, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 742, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 248, 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000905353', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510'}, {'biotype': 'protein_coding', 'protein_start': 248, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'variant_allele': 'A', 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'cds_end': 742, 'strand': -1, 'cdna_end': 886, 'sift_score': 0, 'cdna_start': 886, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 742, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 248, 'blosum62': -3, 'cadd_raw': 5.203168, 'impact': 'MODERATE', 'transcript_id': 'ENST00000923566', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510'}, {'biotype': 'protein_coding', 'protein_start': 248, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'variant_allele': 'A', 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'cds_end': 742, 'strand': -1, 'cdna_end': 1005, 'cdna_start': 1005, 'sift_score': 0, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 742, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 248, 'blosum62': -3, 'cadd_raw': 5.203168, 'impact': 'MODERATE', 'transcript_id': 'ENST00000923567', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510'}, {'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'impact': 'MODERATE', 'blosum62': -3, 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000923568', 'protein_end': 248, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'gene_symbol': 'TP53', 'sift_score': 0, 'cdna_start': 926, 'sift_prediction': 'deleterious', 'cds_start': 742, 'strand': -1, 'cdna_end': 926, 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'variant_allele': 'A', 'cds_end': 742, 'biotype': 'protein_coding', 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 248}, {'biotype': 'protein_coding', 'protein_start': 248, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'variant_allele': 'A', 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'cds_end': 742, 'strand': -1, 'cdna_end': 925, 'cdna_start': 925, 'sift_score': 0, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 742, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 248, 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000923569', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510'}, {'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 248, 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000949117', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'biotype': 'protein_coding', 'protein_start': 248, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'variant_allele': 'A', 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'cds_end': 742, 'strand': -1, 'cdna_end': 884, 'cdna_start': 884, 'gene_symbol': 'TP53', 'cds_start': 742}], 'id': '.', 'start': 7674221, 'most_severe_consequence': 'missense_variant'}]\n"
     ]
    }
   ],
   "source": [
    "import requests\n",
    "import json\n",
    "\n",
    "server = \"https://rest.ensembl.org\" \n",
    "ext = \"/vep/homo_sapiens/region\"\n",
    "\n",
    "headers = { \n",
    "    \"Content-Type\": \"application/json\",\n",
    "    \"Accept\": \"application/json\"\n",
    "} \n",
    "\n",
    "variant = _test[:1]  # 50 varijanti test\n",
    "\n",
    "data = {\n",
    "    \"variants\": variant\n",
    "} \n",
    "param ={ \n",
    "    'canonical':1, # returns canonical transkript \n",
    "    'CADD':1, \n",
    "    'AlphaMissense':1,\n",
    "    'Blosum62':1,\n",
    "    'LoF':1,\n",
    "    'REVEL':1\n",
    "}\n",
    "try:\n",
    "    r = requests.post(server + ext, headers=headers, json=data,params=param)\n",
    "    print(f\"Connection status: {r.status_code}\")\n",
    "except:\n",
    "    ConnectionError(f\"{r.status_code}\")\n",
    "\n",
    "print(r.json())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "8e9a8f90",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'assembly_name': 'GRCh38', 'seq_region_name': '17', 'strand': 1, 'end': 7674221, 'input': '17 7674221 . G A', 'colocated_variants': [{'strand': 1, 'seq_region_name': '17', 'id': 'CM010465', 'start': 7674221, 'clinvar_somatic_classification': 'Neoplasm(oncogenicity:Oncogenic)', 'allele_string': 'HGMD_MUTATION', 'phenotype_or_disease': 1, 'end': 7674221}, {'seq_region_name': '17', 'id': 'CM900211', 'strand': 1, 'end': 7674221, 'phenotype_or_disease': 1, 'allele_string': 'HGMD_MUTATION', 'clinvar_somatic_classification': 'Neoplasm(oncogenicity:Oncogenic)', 'start': 7674221}, {'seq_region_name': '17', 'var_synonyms': {'COSMIC': ['COSM10656']}, 'id': 'COSV52662035', 'strand': 1, 'end': 7674221, 'somatic': 1, 'allele_string': 'COSMIC_MUTATION', 'phenotype_or_disease': 1, 'start': 7674221, 'clinvar_somatic_classification': 'Neoplasm(oncogenicity:Oncogenic)'}, {'somatic': 1, 'end': 7674221, 'phenotype_or_disease': 1, 'allele_string': 'COSMIC_MUTATION', 'start': 7674221, 'clinvar_somatic_classification': 'Neoplasm(oncogenicity:Oncogenic)', 'id': 'COSV52797251', 'var_synonyms': {'COSMIC': ['COSM11564']}, 'seq_region_name': '17', 'strand': 1}, {'strand': 1, 'seq_region_name': '17', 'var_synonyms': {'COSMIC': ['COSM44920']}, 'id': 'COSV53097881', 'phenotype_or_disease': 1, 'allele_string': 'COSMIC_MUTATION', 'clinvar_somatic_classification': 'Neoplasm(oncogenicity:Oncogenic)', 'start': 7674221, 'end': 7674221, 'somatic': 1}, {'end': 7674221, 'frequencies': {'A': {'gnomadg_afr': 0, 'gnomade': 6.157e-06, 'gnomadg': 0, 'gnomade_amr': 2.236e-05, 'gnomadg_asj': 0, 'gnomadg_ami': 0, 'gnomadg_sas': 0, 'gnomade_mid': 0, 'gnomade_asj': 0, 'gnomadg_nfe': 0, 'gnomade_afr': 0, 'gnomadg_fin': 0, 'gnomade_sas': 1.159e-05, 'gnomadg_eas': 0, 'gnomadg_remaining': 0, 'gnomadg_amr': 0, 'gnomade_remaining': 0, 'gnomade_fin': 0, 'gnomade_eas': 0, 'gnomadg_mid': 0, 'gnomade_nfe': 6.296e-06}}, 'clinvar_somatic_classification': 'Neoplasm(oncogenicity:Oncogenic)', 'start': 7674221, 'allele_string': 'G/A/C/T', 'phenotype_or_disease': 1, 'clin_sig_allele': 'T:likely_benign;A:pathogenic/likely_pathogenic;C:pathogenic;A:uncertain_significance;C:likely_pathogenic;A:pathogenic', 'var_synonyms': {'OMIM': [191170.0001], 'PhenCode': ['TP53_g.13379C>T'], 'ClinVar': ['RCV000444845', 'RCV000442243', 'RCV000441557', 'RCV000441091', 'RCV000440560', 'RCV000440422', 'RCV000438698', 'RCV000436398', 'RCV000435803', 'RCV000434504', 'RCV000433905', 'RCV000431689', 'RCV000431508', 'RCV000430543', 'RCV000429884', 'RCV000429777', 'RCV000425682', 'RCV000425083', 'RCV000424415', 'RCV000424308', 'RCV000423804', 'RCV000423184', 'RCV000419857', 'RCV000419150', 'RCV000419032', 'RCV000418495', 'RCV000213057', 'RCV000168242', 'RCV000115735', 'RCV000013140', 'VCV000012347', 'RCV000785485', 'RCV000735293', 'RCV001374442', 'RCV001271100', 'RCV001255674', 'RCV003162243', 'RCV003149569', 'RCV003105772', 'RCV000626118', 'RCV000444519', 'RCV000444427', 'RCV000444356', 'RCV000441711', 'RCV000441010', 'RCV000440334', 'RCV000437882', 'RCV000436038', 'RCV000435353', 'RCV000435050', 'RCV000433611', 'RCV000432931', 'RCV000432207', 'RCV000633396', 'RCV000430735', 'RCV000427948', 'RCV000427544', 'RCV000425782', 'RCV000425394', 'RCV000425100', 'RCV000424776', 'RCV000423297', 'RCV000422668', 'RCV000420498', 'RCV000418362', 'RCV000417693', 'VCV000376652', 'RCV001026425', 'RCV004022238', 'RCV001175916', 'VCV000918353', 'RCV003460463', 'RCV003318332', 'RCV004745154'], 'UniPro': []}, 'id': 'rs121912651', 'seq_region_name': '17', 'pubmed': [25741868, 24033266, 26900293, 32293552, 25157968, 26619011, 25105660, 16489069, 17606709, 30744407, 1631137, 1978757, 21343334, 8099841, 8425176, 8527048, 9598730, 9825943, 12826609, 17427234, 20013323, 28664506, 37182128, 34771529, 34178683, 36866106, 35483882, 37391803, 38310289, 38413718, 39457600, 39208653, 39488539], 'clin_sig': ['uncertain_significance', 'likely_benign', 'likely_pathogenic', 'pathogenic'], 'strand': 1}], 'allele_string': 'G/A', 'transcript_consequences': [{'strand': -1, 'cdna_end': 884, 'gene_symbol': 'TP53', 'sift_score': 0, 'cdna_start': 884, 'sift_prediction': 'deleterious', 'cds_start': 742, 'canonical': 1, 'biotype': 'protein_coding', 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'protein_start': 248, 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'variant_allele': 'A', 'cds_end': 742, 'cadd_raw': 5.203168, 'impact': 'MODERATE', 'blosum62': -3, 'transcript_id': 'ENST00000269305', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'cadd_phred': 29.2, 'revel': 0.927, 'codons': 'Cgg/Tgg', 'protein_end': 248}, {'cadd_raw': 5.203168, 'blosum62': -3, 'impact': 'MODERATE', 'transcript_id': 'ENST00000359597', 'gene_symbol_source': 'HGNC', 'polyphen_score': 1, 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'revel': 0.927, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'protein_end': 248, 'strand': -1, 'cdna_end': 742, 'gene_symbol': 'TP53', 'sift_score': 0, 'cdna_start': 742, 'cds_start': 742, 'polyphen_prediction': 'probably_damaging', 'sift_prediction': 'deleterious', 'biotype': 'protein_coding', 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 248, 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'variant_allele': 'A', 'cds_end': 742}, {'amino_acids': 'R/W', 'polyphen_score': 1, 'gene_id': 'ENSG00000141510', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000413465', 'cadd_raw': 5.203168, 'blosum62': -3, 'impact': 'MODERATE', 'protein_end': 248, 'codons': 'Cgg/Tgg', 'revel': 0.927, 'cadd_phred': 29.2, 'polyphen_prediction': 'probably_damaging', 'sift_prediction': 'deleterious', 'cds_start': 742, 'cdna_start': 742, 'sift_score': 0, 'gene_symbol': 'TP53', 'cdna_end': 742, 'strand': -1, 'cds_end': 742, 'variant_allele': 'A', 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'protein_start': 248, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'biotype': 'protein_coding'}, {'transcript_id': 'ENST00000420246', 'blosum62': -3, 'cadd_raw': 5.203168, 'impact': 'MODERATE', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'gene_symbol_source': 'HGNC', 'revel': 0.927, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'protein_end': 248, 'cdna_end': 875, 'strand': -1, 'sift_prediction': 'deleterious', 'cds_start': 742, 'gene_symbol': 'TP53', 'sift_score': 0, 'cdna_start': 875, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'protein_start': 248, 'biotype': 'protein_coding', 'cds_end': 742, 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'variant_allele': 'A'}, {'revel': 0.927, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'protein_end': 248, 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000445888', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'biotype': 'protein_coding', 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 248, 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'variant_allele': 'A', 'cds_end': 742, 'strand': -1, 'cdna_end': 878, 'gene_symbol': 'TP53', 'cdna_start': 878, 'sift_score': 0, 'sift_prediction': 'deleterious', 'cds_start': 742}, {'gene_symbol': 'TP53', 'cdna_start': 875, 'sift_score': 0, 'sift_prediction': 'deleterious', 'cds_start': 742, 'strand': -1, 'cdna_end': 875, 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'variant_allele': 'A', 'cds_end': 742, 'biotype': 'protein_coding', 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 248, 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000455263', 'protein_end': 248, 'cadd_phred': 29.2, 'revel': 0.927, 'codons': 'Cgg/Tgg'}, {'cdna_end': 924, 'strand': -1, 'sift_prediction': 'deleterious', 'cds_start': 742, 'gene_symbol': 'TP53', 'cdna_start': 924, 'sift_score': 0, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'protein_start': 248, 'biotype': 'protein_coding', 'cds_end': 742, 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'variant_allele': 'A', 'transcript_id': 'ENST00000503591', 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'blosum62': -3, 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'gene_symbol_source': 'HGNC', 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'protein_end': 248}, {'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 116, 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000504290', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'biotype': 'protein_coding', 'protein_start': 116, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'variant_allele': 'A', 'cds_end': 346, 'strand': -1, 'cdna_end': 624, 'cdna_start': 624, 'sift_score': 0, 'gene_symbol': 'TP53', 'cds_start': 346, 'sift_prediction': 'deleterious'}, {'cds_end': 346, 'variant_allele': 'A', 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'protein_start': 116, 'biotype': 'protein_coding', 'cds_start': 346, 'sift_prediction': 'deleterious', 'gene_symbol': 'TP53', 'cdna_start': 624, 'sift_score': 0, 'cdna_end': 624, 'strand': -1, 'protein_end': 116, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000504937', 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'blosum62': -3}, {'cadd_phred': 29.2, 'biotype': 'retained_intron', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:11998', 'variant_allele': 'A', 'impact': 'MODIFIER', 'cadd_raw': 5.203168, 'strand': -1, 'transcript_id': 'ENST00000505014', 'gene_symbol': 'TP53', 'distance': 305, 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510'}, {'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 248, 'biotype': 'protein_coding', 'cds_end': 742, 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'variant_allele': 'A', 'cdna_end': 1024, 'strand': -1, 'cds_start': 742, 'sift_prediction': 'deleterious', 'gene_symbol': 'TP53', 'cdna_start': 1024, 'sift_score': 0, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'protein_end': 248, 'transcript_id': 'ENST00000508793', 'impact': 'MODERATE', 'blosum62': -3, 'cadd_raw': 5.203168, 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'gene_symbol_source': 'HGNC'}, {'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000509690', 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'blosum62': -3, 'protein_end': 116, 'revel': 0.927, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'sift_prediction': 'deleterious', 'cds_start': 346, 'gene_symbol': 'TP53', 'sift_score': 0, 'cdna_start': 478, 'cdna_end': 478, 'strand': -1, 'cds_end': 346, 'variant_allele': 'A', 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 116, 'biotype': 'protein_coding'}, {'biotype': 'protein_coding', 'protein_start': 116, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'variant_allele': 'A', 'cds_end': 346, 'strand': -1, 'cdna_end': 624, 'cdna_start': 624, 'sift_score': 0, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 346, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 116, 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000510385', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510'}, {'cdna_end': 542, 'strand': -1, 'sift_prediction': 'deleterious', 'polyphen_prediction': 'probably_damaging', 'cds_start': 463, 'cdna_start': 542, 'sift_score': 0, 'gene_symbol': 'TP53', 'protein_start': 155, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'biotype': 'protein_coding', 'cds_end': 463, 'variant_allele': 'A', 'transcript_id': 'ENST00000514944', 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'blosum62': -3, 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'polyphen_score': 1, 'gene_symbol_source': 'HGNC', 'codons': 'Cgg/Tgg', 'revel': 0.927, 'cadd_phred': 29.2, 'protein_end': 155}, {'variant_allele': 'A', 'biotype': 'protein_coding_CDS_not_defined', 'cadd_phred': 29.2, 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:11998', 'distance': 33, 'gene_symbol_source': 'HGNC', 'gene_symbol': 'TP53', 'gene_id': 'ENSG00000141510', 'impact': 'MODIFIER', 'cadd_raw': 5.203168, 'strand': -1, 'transcript_id': 'ENST00000574684'}, {'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'impact': 'MODERATE', 'blosum62': -3, 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000576024', 'protein_end': 248, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'gene_symbol': 'TP53', 'sift_score': 0, 'cdna_start': 905, 'sift_prediction': 'deleterious', 'cds_start': 742, 'strand': -1, 'cdna_end': 905, 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'variant_allele': 'A', 'cds_end': 742, 'biotype': 'protein_coding', 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 248}, {'variant_allele': 'A', 'cds_end': 721, 'biotype': 'protein_coding', 'protein_start': 241, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'cdna_start': 860, 'sift_score': 0, 'gene_symbol': 'TP53', 'polyphen_prediction': 'probably_damaging', 'sift_prediction': 'deleterious', 'cds_start': 721, 'strand': -1, 'cdna_end': 860, 'protein_end': 241, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'polyphen_score': 0.999, 'cadd_raw': 5.203168, 'blosum62': -3, 'impact': 'MODERATE', 'transcript_id': 'ENST00000604348'}, {'protein_start': 209, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'biotype': 'protein_coding', 'cds_end': 625, 'variant_allele': 'A', 'cdna_end': 992, 'strand': -1, 'sift_prediction': 'deleterious', 'cds_start': 625, 'sift_score': 0, 'cdna_start': 992, 'gene_symbol': 'TP53', 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 209, 'transcript_id': 'ENST00000610292', 'cadd_raw': 5.203168, 'blosum62': -3, 'impact': 'MODERATE', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'gene_symbol_source': 'HGNC'}, {'transcript_id': 'ENST00000610538', 'cadd_raw': 5.203168, 'impact': 'MODERATE', 'blosum62': -3, 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'gene_symbol_source': 'HGNC', 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 209, 'cdna_end': 875, 'strand': -1, 'sift_prediction': 'deleterious', 'cds_start': 625, 'sift_score': 0, 'cdna_start': 875, 'gene_symbol': 'TP53', 'protein_start': 209, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'biotype': 'protein_coding', 'cds_end': 625, 'variant_allele': 'A'}, {'cds_end': 265, 'variant_allele': 'A', 'protein_start': 89, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'biotype': 'protein_coding', 'cds_start': 265, 'sift_prediction': 'deleterious', 'cdna_start': 624, 'sift_score': 0, 'gene_symbol': 'TP53', 'cdna_end': 624, 'strand': -1, 'protein_end': 89, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000610623', 'blosum62': -3, 'cadd_raw': 5.203168, 'impact': 'MODERATE'}, {'variant_allele': 'A', 'cds_end': 265, 'biotype': 'protein_coding', 'protein_start': 89, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'sift_score': 0, 'cdna_start': 624, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 265, 'strand': -1, 'cdna_end': 624, 'protein_end': 89, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'cadd_raw': 5.203168, 'blosum62': -3, 'impact': 'MODERATE', 'transcript_id': 'ENST00000618944'}, {'sift_prediction': 'deleterious', 'cds_start': 265, 'gene_symbol': 'TP53', 'cdna_start': 624, 'sift_score': 0, 'cdna_end': 624, 'strand': -1, 'cds_end': 265, 'variant_allele': 'A', 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'protein_start': 89, 'biotype': 'protein_coding', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000619186', 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'protein_end': 89, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg'}, {'cds_end': 625, 'variant_allele': 'A', 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 209, 'biotype': 'protein_coding', 'sift_prediction': 'deleterious', 'cds_start': 625, 'gene_symbol': 'TP53', 'cdna_start': 878, 'sift_score': 0, 'cdna_end': 878, 'strand': -1, 'protein_end': 209, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000619485', 'cadd_raw': 5.203168, 'impact': 'MODERATE', 'blosum62': -3}, {'cadd_raw': 5.203168, 'impact': 'MODERATE', 'blosum62': -3, 'transcript_id': 'ENST00000620739', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'protein_end': 209, 'strand': -1, 'cdna_end': 932, 'gene_symbol': 'TP53', 'cdna_start': 932, 'sift_score': 0, 'sift_prediction': 'deleterious', 'cds_start': 625, 'biotype': 'protein_coding', 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'protein_start': 209, 'variant_allele': 'A', 'cds_end': 625}, {'protein_end': 209, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'blosum62': -3, 'cadd_raw': 5.203168, 'impact': 'MODERATE', 'transcript_id': 'ENST00000622645', 'variant_allele': 'A', 'cds_end': 625, 'biotype': 'protein_coding', 'protein_start': 209, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'sift_score': 0, 'cdna_start': 875, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 625, 'strand': -1, 'cdna_end': 875}, {'cds_end': 625, 'variant_allele': 'A', 'protein_start': 209, 'consequence_terms': ['missense_variant', 'NMD_transcript_variant'], 'hgnc_id': 'HGNC:11998', 'biotype': 'nonsense_mediated_decay', 'polyphen_prediction': 'probably_damaging', 'sift_prediction': 'deleterious_low_confidence', 'cds_start': 625, 'cdna_start': 885, 'sift_score': 0, 'gene_symbol': 'TP53', 'cdna_end': 885, 'strand': -1, 'protein_end': 209, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'amino_acids': 'R/W', 'polyphen_score': 1, 'gene_id': 'ENSG00000141510', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000635293', 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168}, {'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 209, 'cadd_raw': 5.203168, 'blosum62': -3, 'impact': 'MODERATE', 'transcript_id': 'ENST00000714356', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'biotype': 'protein_coding', 'protein_start': 209, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'variant_allele': 'A', 'cds_end': 625, 'strand': -1, 'cdna_end': 1005, 'sift_score': 0, 'cdna_start': 1005, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 625}, {'cds_end': 742, 'variant_allele': 'A', 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'protein_start': 248, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'biotype': 'protein_coding', 'cds_start': 742, 'sift_prediction': 'deleterious', 'cdna_start': 812, 'sift_score': 0, 'gene_symbol': 'TP53', 'cdna_end': 812, 'strand': -1, 'protein_end': 248, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000714357', 'cadd_raw': 5.203168, 'blosum62': -3, 'impact': 'MODERATE'}, {'variant_allele': 'A', 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['3_prime_UTR_variant', 'NMD_transcript_variant'], 'biotype': 'nonsense_mediated_decay', 'cadd_phred': 29.2, 'gene_id': 'ENSG00000141510', 'gene_symbol_source': 'HGNC', 'cdna_start': 862, 'gene_symbol': 'TP53', 'cdna_end': 862, 'transcript_id': 'ENST00000714358', 'cadd_raw': 5.203168, 'impact': 'MODIFIER', 'strand': -1}, {'cdna_start': 859, 'sift_score': 0, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 742, 'strand': -1, 'cdna_end': 859, 'variant_allele': 'A', 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'cds_end': 742, 'biotype': 'protein_coding', 'protein_start': 248, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'blosum62': -3, 'transcript_id': 'ENST00000714359', 'protein_end': 248, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2}, {'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'protein_end': 248, 'impact': 'MODERATE', 'blosum62': -3, 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000714408', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'biotype': 'protein_coding', 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 248, 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'variant_allele': 'A', 'cds_end': 742, 'strand': -1, 'cdna_end': 884, 'gene_symbol': 'TP53', 'cdna_start': 884, 'sift_score': 0, 'cds_start': 742, 'sift_prediction': 'deleterious'}, {'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'variant_allele': 'A', 'cds_end': 742, 'biotype': 'protein_coding', 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'protein_start': 248, 'gene_symbol': 'TP53', 'cdna_start': 884, 'sift_score': 0, 'sift_prediction': 'deleterious', 'cds_start': 742, 'strand': -1, 'cdna_end': 884, 'protein_end': 248, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'blosum62': -3, 'cadd_raw': 5.203168, 'impact': 'MODERATE', 'transcript_id': 'ENST00000714409'}, {'biotype': 'protein_coding', 'protein_start': 248, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'variant_allele': 'A', 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'cds_end': 742, 'strand': -1, 'cdna_end': 837, 'cdna_start': 837, 'sift_score': 0, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 742, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 248, 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000905353', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510'}, {'biotype': 'protein_coding', 'protein_start': 248, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'variant_allele': 'A', 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'cds_end': 742, 'strand': -1, 'cdna_end': 886, 'sift_score': 0, 'cdna_start': 886, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 742, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 248, 'blosum62': -3, 'cadd_raw': 5.203168, 'impact': 'MODERATE', 'transcript_id': 'ENST00000923566', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510'}, {'biotype': 'protein_coding', 'protein_start': 248, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'variant_allele': 'A', 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'cds_end': 742, 'strand': -1, 'cdna_end': 1005, 'cdna_start': 1005, 'sift_score': 0, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 742, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 248, 'blosum62': -3, 'cadd_raw': 5.203168, 'impact': 'MODERATE', 'transcript_id': 'ENST00000923567', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510'}, {'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'impact': 'MODERATE', 'blosum62': -3, 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000923568', 'protein_end': 248, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'gene_symbol': 'TP53', 'sift_score': 0, 'cdna_start': 926, 'sift_prediction': 'deleterious', 'cds_start': 742, 'strand': -1, 'cdna_end': 926, 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'variant_allele': 'A', 'cds_end': 742, 'biotype': 'protein_coding', 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 248}, {'biotype': 'protein_coding', 'protein_start': 248, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'variant_allele': 'A', 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'cds_end': 742, 'strand': -1, 'cdna_end': 925, 'cdna_start': 925, 'sift_score': 0, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 742, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 248, 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000923569', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510'}, {'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 248, 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000949117', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'biotype': 'protein_coding', 'protein_start': 248, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'variant_allele': 'A', 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'cds_end': 742, 'strand': -1, 'cdna_end': 884, 'cdna_start': 884, 'gene_symbol': 'TP53', 'cds_start': 742}], 'id': '.', 'start': 7674221, 'most_severe_consequence': 'missense_variant'}\n"
     ]
    }
   ],
   "source": [
    "result = r.json()\n",
    "print(result[0]) #json string"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "3f1ae84f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "assembly_name GRCh38\n",
      "seq_region_name 17\n",
      "strand 1\n",
      "end 7674221\n",
      "input 17 7674221 . G A\n",
      "colocated_variants [{'strand': 1, 'seq_region_name': '17', 'id': 'CM010465', 'start': 7674221, 'clinvar_somatic_classification': 'Neoplasm(oncogenicity:Oncogenic)', 'allele_string': 'HGMD_MUTATION', 'phenotype_or_disease': 1, 'end': 7674221}, {'seq_region_name': '17', 'id': 'CM900211', 'strand': 1, 'end': 7674221, 'phenotype_or_disease': 1, 'allele_string': 'HGMD_MUTATION', 'clinvar_somatic_classification': 'Neoplasm(oncogenicity:Oncogenic)', 'start': 7674221}, {'seq_region_name': '17', 'var_synonyms': {'COSMIC': ['COSM10656']}, 'id': 'COSV52662035', 'strand': 1, 'end': 7674221, 'somatic': 1, 'allele_string': 'COSMIC_MUTATION', 'phenotype_or_disease': 1, 'start': 7674221, 'clinvar_somatic_classification': 'Neoplasm(oncogenicity:Oncogenic)'}, {'somatic': 1, 'end': 7674221, 'phenotype_or_disease': 1, 'allele_string': 'COSMIC_MUTATION', 'start': 7674221, 'clinvar_somatic_classification': 'Neoplasm(oncogenicity:Oncogenic)', 'id': 'COSV52797251', 'var_synonyms': {'COSMIC': ['COSM11564']}, 'seq_region_name': '17', 'strand': 1}, {'strand': 1, 'seq_region_name': '17', 'var_synonyms': {'COSMIC': ['COSM44920']}, 'id': 'COSV53097881', 'phenotype_or_disease': 1, 'allele_string': 'COSMIC_MUTATION', 'clinvar_somatic_classification': 'Neoplasm(oncogenicity:Oncogenic)', 'start': 7674221, 'end': 7674221, 'somatic': 1}, {'end': 7674221, 'frequencies': {'A': {'gnomadg_afr': 0, 'gnomade': 6.157e-06, 'gnomadg': 0, 'gnomade_amr': 2.236e-05, 'gnomadg_asj': 0, 'gnomadg_ami': 0, 'gnomadg_sas': 0, 'gnomade_mid': 0, 'gnomade_asj': 0, 'gnomadg_nfe': 0, 'gnomade_afr': 0, 'gnomadg_fin': 0, 'gnomade_sas': 1.159e-05, 'gnomadg_eas': 0, 'gnomadg_remaining': 0, 'gnomadg_amr': 0, 'gnomade_remaining': 0, 'gnomade_fin': 0, 'gnomade_eas': 0, 'gnomadg_mid': 0, 'gnomade_nfe': 6.296e-06}}, 'clinvar_somatic_classification': 'Neoplasm(oncogenicity:Oncogenic)', 'start': 7674221, 'allele_string': 'G/A/C/T', 'phenotype_or_disease': 1, 'clin_sig_allele': 'T:likely_benign;A:pathogenic/likely_pathogenic;C:pathogenic;A:uncertain_significance;C:likely_pathogenic;A:pathogenic', 'var_synonyms': {'OMIM': [191170.0001], 'PhenCode': ['TP53_g.13379C>T'], 'ClinVar': ['RCV000444845', 'RCV000442243', 'RCV000441557', 'RCV000441091', 'RCV000440560', 'RCV000440422', 'RCV000438698', 'RCV000436398', 'RCV000435803', 'RCV000434504', 'RCV000433905', 'RCV000431689', 'RCV000431508', 'RCV000430543', 'RCV000429884', 'RCV000429777', 'RCV000425682', 'RCV000425083', 'RCV000424415', 'RCV000424308', 'RCV000423804', 'RCV000423184', 'RCV000419857', 'RCV000419150', 'RCV000419032', 'RCV000418495', 'RCV000213057', 'RCV000168242', 'RCV000115735', 'RCV000013140', 'VCV000012347', 'RCV000785485', 'RCV000735293', 'RCV001374442', 'RCV001271100', 'RCV001255674', 'RCV003162243', 'RCV003149569', 'RCV003105772', 'RCV000626118', 'RCV000444519', 'RCV000444427', 'RCV000444356', 'RCV000441711', 'RCV000441010', 'RCV000440334', 'RCV000437882', 'RCV000436038', 'RCV000435353', 'RCV000435050', 'RCV000433611', 'RCV000432931', 'RCV000432207', 'RCV000633396', 'RCV000430735', 'RCV000427948', 'RCV000427544', 'RCV000425782', 'RCV000425394', 'RCV000425100', 'RCV000424776', 'RCV000423297', 'RCV000422668', 'RCV000420498', 'RCV000418362', 'RCV000417693', 'VCV000376652', 'RCV001026425', 'RCV004022238', 'RCV001175916', 'VCV000918353', 'RCV003460463', 'RCV003318332', 'RCV004745154'], 'UniPro': []}, 'id': 'rs121912651', 'seq_region_name': '17', 'pubmed': [25741868, 24033266, 26900293, 32293552, 25157968, 26619011, 25105660, 16489069, 17606709, 30744407, 1631137, 1978757, 21343334, 8099841, 8425176, 8527048, 9598730, 9825943, 12826609, 17427234, 20013323, 28664506, 37182128, 34771529, 34178683, 36866106, 35483882, 37391803, 38310289, 38413718, 39457600, 39208653, 39488539], 'clin_sig': ['uncertain_significance', 'likely_benign', 'likely_pathogenic', 'pathogenic'], 'strand': 1}]\n",
      "allele_string G/A\n",
      "transcript_consequences [{'strand': -1, 'cdna_end': 884, 'gene_symbol': 'TP53', 'sift_score': 0, 'cdna_start': 884, 'sift_prediction': 'deleterious', 'cds_start': 742, 'canonical': 1, 'biotype': 'protein_coding', 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'protein_start': 248, 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'variant_allele': 'A', 'cds_end': 742, 'cadd_raw': 5.203168, 'impact': 'MODERATE', 'blosum62': -3, 'transcript_id': 'ENST00000269305', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'cadd_phred': 29.2, 'revel': 0.927, 'codons': 'Cgg/Tgg', 'protein_end': 248}, {'cadd_raw': 5.203168, 'blosum62': -3, 'impact': 'MODERATE', 'transcript_id': 'ENST00000359597', 'gene_symbol_source': 'HGNC', 'polyphen_score': 1, 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'revel': 0.927, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'protein_end': 248, 'strand': -1, 'cdna_end': 742, 'gene_symbol': 'TP53', 'sift_score': 0, 'cdna_start': 742, 'cds_start': 742, 'polyphen_prediction': 'probably_damaging', 'sift_prediction': 'deleterious', 'biotype': 'protein_coding', 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 248, 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'variant_allele': 'A', 'cds_end': 742}, {'amino_acids': 'R/W', 'polyphen_score': 1, 'gene_id': 'ENSG00000141510', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000413465', 'cadd_raw': 5.203168, 'blosum62': -3, 'impact': 'MODERATE', 'protein_end': 248, 'codons': 'Cgg/Tgg', 'revel': 0.927, 'cadd_phred': 29.2, 'polyphen_prediction': 'probably_damaging', 'sift_prediction': 'deleterious', 'cds_start': 742, 'cdna_start': 742, 'sift_score': 0, 'gene_symbol': 'TP53', 'cdna_end': 742, 'strand': -1, 'cds_end': 742, 'variant_allele': 'A', 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'protein_start': 248, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'biotype': 'protein_coding'}, {'transcript_id': 'ENST00000420246', 'blosum62': -3, 'cadd_raw': 5.203168, 'impact': 'MODERATE', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'gene_symbol_source': 'HGNC', 'revel': 0.927, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'protein_end': 248, 'cdna_end': 875, 'strand': -1, 'sift_prediction': 'deleterious', 'cds_start': 742, 'gene_symbol': 'TP53', 'sift_score': 0, 'cdna_start': 875, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'protein_start': 248, 'biotype': 'protein_coding', 'cds_end': 742, 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'variant_allele': 'A'}, {'revel': 0.927, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'protein_end': 248, 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000445888', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'biotype': 'protein_coding', 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 248, 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'variant_allele': 'A', 'cds_end': 742, 'strand': -1, 'cdna_end': 878, 'gene_symbol': 'TP53', 'cdna_start': 878, 'sift_score': 0, 'sift_prediction': 'deleterious', 'cds_start': 742}, {'gene_symbol': 'TP53', 'cdna_start': 875, 'sift_score': 0, 'sift_prediction': 'deleterious', 'cds_start': 742, 'strand': -1, 'cdna_end': 875, 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'variant_allele': 'A', 'cds_end': 742, 'biotype': 'protein_coding', 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 248, 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000455263', 'protein_end': 248, 'cadd_phred': 29.2, 'revel': 0.927, 'codons': 'Cgg/Tgg'}, {'cdna_end': 924, 'strand': -1, 'sift_prediction': 'deleterious', 'cds_start': 742, 'gene_symbol': 'TP53', 'cdna_start': 924, 'sift_score': 0, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'protein_start': 248, 'biotype': 'protein_coding', 'cds_end': 742, 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'variant_allele': 'A', 'transcript_id': 'ENST00000503591', 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'blosum62': -3, 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'gene_symbol_source': 'HGNC', 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'protein_end': 248}, {'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 116, 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000504290', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'biotype': 'protein_coding', 'protein_start': 116, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'variant_allele': 'A', 'cds_end': 346, 'strand': -1, 'cdna_end': 624, 'cdna_start': 624, 'sift_score': 0, 'gene_symbol': 'TP53', 'cds_start': 346, 'sift_prediction': 'deleterious'}, {'cds_end': 346, 'variant_allele': 'A', 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'protein_start': 116, 'biotype': 'protein_coding', 'cds_start': 346, 'sift_prediction': 'deleterious', 'gene_symbol': 'TP53', 'cdna_start': 624, 'sift_score': 0, 'cdna_end': 624, 'strand': -1, 'protein_end': 116, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000504937', 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'blosum62': -3}, {'cadd_phred': 29.2, 'biotype': 'retained_intron', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:11998', 'variant_allele': 'A', 'impact': 'MODIFIER', 'cadd_raw': 5.203168, 'strand': -1, 'transcript_id': 'ENST00000505014', 'gene_symbol': 'TP53', 'distance': 305, 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510'}, {'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 248, 'biotype': 'protein_coding', 'cds_end': 742, 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'variant_allele': 'A', 'cdna_end': 1024, 'strand': -1, 'cds_start': 742, 'sift_prediction': 'deleterious', 'gene_symbol': 'TP53', 'cdna_start': 1024, 'sift_score': 0, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'protein_end': 248, 'transcript_id': 'ENST00000508793', 'impact': 'MODERATE', 'blosum62': -3, 'cadd_raw': 5.203168, 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'gene_symbol_source': 'HGNC'}, {'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000509690', 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'blosum62': -3, 'protein_end': 116, 'revel': 0.927, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'sift_prediction': 'deleterious', 'cds_start': 346, 'gene_symbol': 'TP53', 'sift_score': 0, 'cdna_start': 478, 'cdna_end': 478, 'strand': -1, 'cds_end': 346, 'variant_allele': 'A', 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 116, 'biotype': 'protein_coding'}, {'biotype': 'protein_coding', 'protein_start': 116, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'variant_allele': 'A', 'cds_end': 346, 'strand': -1, 'cdna_end': 624, 'cdna_start': 624, 'sift_score': 0, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 346, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 116, 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000510385', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510'}, {'cdna_end': 542, 'strand': -1, 'sift_prediction': 'deleterious', 'polyphen_prediction': 'probably_damaging', 'cds_start': 463, 'cdna_start': 542, 'sift_score': 0, 'gene_symbol': 'TP53', 'protein_start': 155, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'biotype': 'protein_coding', 'cds_end': 463, 'variant_allele': 'A', 'transcript_id': 'ENST00000514944', 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'blosum62': -3, 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'polyphen_score': 1, 'gene_symbol_source': 'HGNC', 'codons': 'Cgg/Tgg', 'revel': 0.927, 'cadd_phred': 29.2, 'protein_end': 155}, {'variant_allele': 'A', 'biotype': 'protein_coding_CDS_not_defined', 'cadd_phred': 29.2, 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:11998', 'distance': 33, 'gene_symbol_source': 'HGNC', 'gene_symbol': 'TP53', 'gene_id': 'ENSG00000141510', 'impact': 'MODIFIER', 'cadd_raw': 5.203168, 'strand': -1, 'transcript_id': 'ENST00000574684'}, {'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'impact': 'MODERATE', 'blosum62': -3, 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000576024', 'protein_end': 248, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'gene_symbol': 'TP53', 'sift_score': 0, 'cdna_start': 905, 'sift_prediction': 'deleterious', 'cds_start': 742, 'strand': -1, 'cdna_end': 905, 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'variant_allele': 'A', 'cds_end': 742, 'biotype': 'protein_coding', 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 248}, {'variant_allele': 'A', 'cds_end': 721, 'biotype': 'protein_coding', 'protein_start': 241, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'cdna_start': 860, 'sift_score': 0, 'gene_symbol': 'TP53', 'polyphen_prediction': 'probably_damaging', 'sift_prediction': 'deleterious', 'cds_start': 721, 'strand': -1, 'cdna_end': 860, 'protein_end': 241, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'polyphen_score': 0.999, 'cadd_raw': 5.203168, 'blosum62': -3, 'impact': 'MODERATE', 'transcript_id': 'ENST00000604348'}, {'protein_start': 209, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'biotype': 'protein_coding', 'cds_end': 625, 'variant_allele': 'A', 'cdna_end': 992, 'strand': -1, 'sift_prediction': 'deleterious', 'cds_start': 625, 'sift_score': 0, 'cdna_start': 992, 'gene_symbol': 'TP53', 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 209, 'transcript_id': 'ENST00000610292', 'cadd_raw': 5.203168, 'blosum62': -3, 'impact': 'MODERATE', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'gene_symbol_source': 'HGNC'}, {'transcript_id': 'ENST00000610538', 'cadd_raw': 5.203168, 'impact': 'MODERATE', 'blosum62': -3, 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'gene_symbol_source': 'HGNC', 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 209, 'cdna_end': 875, 'strand': -1, 'sift_prediction': 'deleterious', 'cds_start': 625, 'sift_score': 0, 'cdna_start': 875, 'gene_symbol': 'TP53', 'protein_start': 209, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'biotype': 'protein_coding', 'cds_end': 625, 'variant_allele': 'A'}, {'cds_end': 265, 'variant_allele': 'A', 'protein_start': 89, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'biotype': 'protein_coding', 'cds_start': 265, 'sift_prediction': 'deleterious', 'cdna_start': 624, 'sift_score': 0, 'gene_symbol': 'TP53', 'cdna_end': 624, 'strand': -1, 'protein_end': 89, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000610623', 'blosum62': -3, 'cadd_raw': 5.203168, 'impact': 'MODERATE'}, {'variant_allele': 'A', 'cds_end': 265, 'biotype': 'protein_coding', 'protein_start': 89, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'sift_score': 0, 'cdna_start': 624, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 265, 'strand': -1, 'cdna_end': 624, 'protein_end': 89, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'cadd_raw': 5.203168, 'blosum62': -3, 'impact': 'MODERATE', 'transcript_id': 'ENST00000618944'}, {'sift_prediction': 'deleterious', 'cds_start': 265, 'gene_symbol': 'TP53', 'cdna_start': 624, 'sift_score': 0, 'cdna_end': 624, 'strand': -1, 'cds_end': 265, 'variant_allele': 'A', 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'protein_start': 89, 'biotype': 'protein_coding', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000619186', 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'protein_end': 89, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg'}, {'cds_end': 625, 'variant_allele': 'A', 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 209, 'biotype': 'protein_coding', 'sift_prediction': 'deleterious', 'cds_start': 625, 'gene_symbol': 'TP53', 'cdna_start': 878, 'sift_score': 0, 'cdna_end': 878, 'strand': -1, 'protein_end': 209, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000619485', 'cadd_raw': 5.203168, 'impact': 'MODERATE', 'blosum62': -3}, {'cadd_raw': 5.203168, 'impact': 'MODERATE', 'blosum62': -3, 'transcript_id': 'ENST00000620739', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'protein_end': 209, 'strand': -1, 'cdna_end': 932, 'gene_symbol': 'TP53', 'cdna_start': 932, 'sift_score': 0, 'sift_prediction': 'deleterious', 'cds_start': 625, 'biotype': 'protein_coding', 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'protein_start': 209, 'variant_allele': 'A', 'cds_end': 625}, {'protein_end': 209, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'blosum62': -3, 'cadd_raw': 5.203168, 'impact': 'MODERATE', 'transcript_id': 'ENST00000622645', 'variant_allele': 'A', 'cds_end': 625, 'biotype': 'protein_coding', 'protein_start': 209, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'sift_score': 0, 'cdna_start': 875, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 625, 'strand': -1, 'cdna_end': 875}, {'cds_end': 625, 'variant_allele': 'A', 'protein_start': 209, 'consequence_terms': ['missense_variant', 'NMD_transcript_variant'], 'hgnc_id': 'HGNC:11998', 'biotype': 'nonsense_mediated_decay', 'polyphen_prediction': 'probably_damaging', 'sift_prediction': 'deleterious_low_confidence', 'cds_start': 625, 'cdna_start': 885, 'sift_score': 0, 'gene_symbol': 'TP53', 'cdna_end': 885, 'strand': -1, 'protein_end': 209, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'amino_acids': 'R/W', 'polyphen_score': 1, 'gene_id': 'ENSG00000141510', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000635293', 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168}, {'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 209, 'cadd_raw': 5.203168, 'blosum62': -3, 'impact': 'MODERATE', 'transcript_id': 'ENST00000714356', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'biotype': 'protein_coding', 'protein_start': 209, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'variant_allele': 'A', 'cds_end': 625, 'strand': -1, 'cdna_end': 1005, 'sift_score': 0, 'cdna_start': 1005, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 625}, {'cds_end': 742, 'variant_allele': 'A', 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'protein_start': 248, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'biotype': 'protein_coding', 'cds_start': 742, 'sift_prediction': 'deleterious', 'cdna_start': 812, 'sift_score': 0, 'gene_symbol': 'TP53', 'cdna_end': 812, 'strand': -1, 'protein_end': 248, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000714357', 'cadd_raw': 5.203168, 'blosum62': -3, 'impact': 'MODERATE'}, {'variant_allele': 'A', 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['3_prime_UTR_variant', 'NMD_transcript_variant'], 'biotype': 'nonsense_mediated_decay', 'cadd_phred': 29.2, 'gene_id': 'ENSG00000141510', 'gene_symbol_source': 'HGNC', 'cdna_start': 862, 'gene_symbol': 'TP53', 'cdna_end': 862, 'transcript_id': 'ENST00000714358', 'cadd_raw': 5.203168, 'impact': 'MODIFIER', 'strand': -1}, {'cdna_start': 859, 'sift_score': 0, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 742, 'strand': -1, 'cdna_end': 859, 'variant_allele': 'A', 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'cds_end': 742, 'biotype': 'protein_coding', 'protein_start': 248, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'blosum62': -3, 'transcript_id': 'ENST00000714359', 'protein_end': 248, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2}, {'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'protein_end': 248, 'impact': 'MODERATE', 'blosum62': -3, 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000714408', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'biotype': 'protein_coding', 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 248, 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'variant_allele': 'A', 'cds_end': 742, 'strand': -1, 'cdna_end': 884, 'gene_symbol': 'TP53', 'cdna_start': 884, 'sift_score': 0, 'cds_start': 742, 'sift_prediction': 'deleterious'}, {'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'variant_allele': 'A', 'cds_end': 742, 'biotype': 'protein_coding', 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'protein_start': 248, 'gene_symbol': 'TP53', 'cdna_start': 884, 'sift_score': 0, 'sift_prediction': 'deleterious', 'cds_start': 742, 'strand': -1, 'cdna_end': 884, 'protein_end': 248, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'blosum62': -3, 'cadd_raw': 5.203168, 'impact': 'MODERATE', 'transcript_id': 'ENST00000714409'}, {'biotype': 'protein_coding', 'protein_start': 248, 'hgnc_id': 'HGNC:11998', 'consequence_terms': ['missense_variant'], 'variant_allele': 'A', 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'cds_end': 742, 'strand': -1, 'cdna_end': 837, 'cdna_start': 837, 'sift_score': 0, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 742, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 248, 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000905353', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510'}, {'biotype': 'protein_coding', 'protein_start': 248, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'variant_allele': 'A', 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'cds_end': 742, 'strand': -1, 'cdna_end': 886, 'sift_score': 0, 'cdna_start': 886, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 742, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 248, 'blosum62': -3, 'cadd_raw': 5.203168, 'impact': 'MODERATE', 'transcript_id': 'ENST00000923566', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510'}, {'biotype': 'protein_coding', 'protein_start': 248, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'variant_allele': 'A', 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'cds_end': 742, 'strand': -1, 'cdna_end': 1005, 'cdna_start': 1005, 'sift_score': 0, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 742, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 248, 'blosum62': -3, 'cadd_raw': 5.203168, 'impact': 'MODERATE', 'transcript_id': 'ENST00000923567', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510'}, {'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000141510', 'amino_acids': 'R/W', 'impact': 'MODERATE', 'blosum62': -3, 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000923568', 'protein_end': 248, 'cadd_phred': 29.2, 'codons': 'Cgg/Tgg', 'gene_symbol': 'TP53', 'sift_score': 0, 'cdna_start': 926, 'sift_prediction': 'deleterious', 'cds_start': 742, 'strand': -1, 'cdna_end': 926, 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'variant_allele': 'A', 'cds_end': 742, 'biotype': 'protein_coding', 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'protein_start': 248}, {'biotype': 'protein_coding', 'protein_start': 248, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'variant_allele': 'A', 'alphamissense': {'am_pathogenicity': 0.9968, 'am_class': 'likely_pathogenic'}, 'cds_end': 742, 'strand': -1, 'cdna_end': 925, 'cdna_start': 925, 'sift_score': 0, 'gene_symbol': 'TP53', 'sift_prediction': 'deleterious', 'cds_start': 742, 'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 248, 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000923569', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510'}, {'codons': 'Cgg/Tgg', 'cadd_phred': 29.2, 'protein_end': 248, 'blosum62': -3, 'impact': 'MODERATE', 'cadd_raw': 5.203168, 'transcript_id': 'ENST00000949117', 'gene_symbol_source': 'HGNC', 'amino_acids': 'R/W', 'gene_id': 'ENSG00000141510', 'biotype': 'protein_coding', 'protein_start': 248, 'consequence_terms': ['missense_variant'], 'hgnc_id': 'HGNC:11998', 'variant_allele': 'A', 'alphamissense': {'am_class': 'likely_pathogenic', 'am_pathogenicity': 0.9968}, 'cds_end': 742, 'strand': -1, 'cdna_end': 884, 'cdna_start': 884, 'gene_symbol': 'TP53', 'cds_start': 742}]\n",
      "id .\n",
      "start 7674221\n",
      "most_severe_consequence missense_variant\n"
     ]
    }
   ],
   "source": [
    "for key,value in result[0].items():\n",
    "    print(key,value) # transcript_consequences = in this nested dict are stored variants information!!!!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "23db7bc5",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "ciljneAnotacije = result[0]['transcript_consequences']\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8b73fb9f",
   "metadata": {},
   "source": [
    "The output contains many annotations that are not needed, so I am only looking at the 'transcript_consequences' nested list, which contains annotations relevant for ML.\n",
    "\n",
    "Also, transcripts contain different keys, which is expected behavior in the context of genomics, as one variant can have multiple transcripts with different functional effects (1 gene = many different products).\n",
    "\n",
    "For this reason, I will make a selection such that 1 Variant : 1 Transcript (representative)\n",
    "\n",
    "> The transcript must be canonical:1 and 'protein-coding'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "6dc83863",
   "metadata": {},
   "outputs": [],
   "source": [
    "# transcripts selection\n",
    "def select_transcript(anotacije):\n",
    "    is_proteinCoding = [p for p in anotacije if p.get('biotype')=='protein_coding'] # only protein coding hence they can be pathogenic\n",
    "    is_canonical = [c for c in anotacije if c.get('canonical')==1] # only canonical\n",
    "    selected = is_canonical[0] if anotacije else is_proteinCoding[0] \n",
    "    return selected\n",
    "selected = select_transcript(ciljneAnotacije)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "794f01e8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[{'variant_id': 0, 'cadd_phred': 29.2, 'cadd_raw': 5.203168, 'revel': 0.927, 'sift_score': 0, 'polyphen_score': None, 'blosum62': -3, 'alphamissense_score': 0.9968, 'impact': 'MODERATE', 'consequence': 'missense_variant'}]\n"
     ]
    }
   ],
   "source": [
    "rows=[]\n",
    "for i in range(len(result)):\n",
    "    transcripts = result[i]['transcript_consequences']\n",
    "    select = select_transcript(transcripts)\n",
    "    features = {\n",
    "    'variant_id': i,\n",
    "    'cadd_phred': select.get('cadd_phred'),\n",
    "    'cadd_raw': select.get('cadd_raw'),\n",
    "    'revel': select.get('revel'),\n",
    "    'sift_score': select.get('sift_score'),\n",
    "    'polyphen_score': select.get('polyphen_score'),\n",
    "    'blosum62': select.get('blosum62'),\n",
    "    'alphamissense_score': select.get('alphamissense',{}).get('am_pathogenicity'), #alphamissens vraca 2 vrednosti ali uzimam samo prvu tj scor, label mi ne treba\n",
    "    'impact': select.get('impact'),\n",
    "    'consequence': (select.get('consequence_terms', [None])[0])\n",
    "    }   \n",
    "\n",
    "    rows.append(features)   \n",
    "print(rows)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c7819cee",
   "metadata": {},
   "source": [
    "#### Making dataset from these features"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "d2d2f2a8",
   "metadata": {},
   "outputs": [],
   "source": [
    "values = []\n",
    "for i in rows:\n",
    "    column_names = i.keys()\n",
    "    column_values = i.values()\n",
    "    values.append(list(column_values))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "c3c74c41",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0, 29.2, 5.203168, 0.927, 0, None, -3, 0.9968, 'MODERATE', 'missense_variant']\n"
     ]
    }
   ],
   "source": [
    "for i in rows:\n",
    "    data = list(i.values())\n",
    "    print(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "1db0a877",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>cadd_phred</th>\n",
       "      <th>cadd_raw</th>\n",
       "      <th>revel</th>\n",
       "      <th>sift_score</th>\n",
       "      <th>polyphen_score</th>\n",
       "      <th>blosum62</th>\n",
       "      <th>alphamissense_score</th>\n",
       "      <th>impact</th>\n",
       "      <th>consequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>29.2</td>\n",
       "      <td>5.203168</td>\n",
       "      <td>0.927</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>-3</td>\n",
       "      <td>0.9968</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>missense_variant</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   variant_id  cadd_phred  cadd_raw  revel  sift_score polyphen_score  \\\n",
       "0           0        29.2  5.203168  0.927           0           None   \n",
       "\n",
       "   blosum62  alphamissense_score    impact       consequence  \n",
       "0        -3               0.9968  MODERATE  missense_variant  "
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "features_df = pd.DataFrame(data=values,columns= column_names)\n",
    "features_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "c1040510",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start</th>\n",
       "      <th>ReferenceAlleleVCF</th>\n",
       "      <th>AlternateAlleleVCF</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>ClinSigSimple</th>\n",
       "      <th>vcf_format</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17</td>\n",
       "      <td>7674221</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1</td>\n",
       "      <td>17 7674221 . G A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chromosome    Start ReferenceAlleleVCF AlternateAlleleVCF GeneSymbol  \\\n",
       "0         17  7674221                  G                  A       TP53   \n",
       "\n",
       "   ClinSigSimple        vcf_format  \n",
       "0              1  17 7674221 . G A  "
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# copy of the starting dataset , 50 rows appending to 50 variants (TEST)\n",
    "dataset_cp = dataset.copy()\n",
    "dataset_cp = dataset_cp.iloc[:1,:]\n",
    "dataset_cp"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4bf0eb62",
   "metadata": {},
   "source": [
    "Ovde je prikazan integrisani dataset nakon spajanja ClinVar zapisa sa Ensembl VEP anotacijama. Svi atributi nisu namenjeni modeliranju, već služe kao izvor ili meta-informacije"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "c79e6302",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start</th>\n",
       "      <th>ReferenceAlleleVCF</th>\n",
       "      <th>AlternateAlleleVCF</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>ClinSigSimple</th>\n",
       "      <th>vcf_format</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>cadd_phred</th>\n",
       "      <th>cadd_raw</th>\n",
       "      <th>revel</th>\n",
       "      <th>sift_score</th>\n",
       "      <th>polyphen_score</th>\n",
       "      <th>blosum62</th>\n",
       "      <th>alphamissense_score</th>\n",
       "      <th>impact</th>\n",
       "      <th>consequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17</td>\n",
       "      <td>7674221</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1</td>\n",
       "      <td>17 7674221 . G A</td>\n",
       "      <td>0</td>\n",
       "      <td>29.2</td>\n",
       "      <td>5.203168</td>\n",
       "      <td>0.927</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>-3</td>\n",
       "      <td>0.9968</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>missense_variant</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chromosome    Start ReferenceAlleleVCF AlternateAlleleVCF GeneSymbol  \\\n",
       "0         17  7674221                  G                  A       TP53   \n",
       "\n",
       "   ClinSigSimple        vcf_format  variant_id  cadd_phred  cadd_raw  revel  \\\n",
       "0              1  17 7674221 . G A           0        29.2  5.203168  0.927   \n",
       "\n",
       "   sift_score polyphen_score  blosum62  alphamissense_score    impact  \\\n",
       "0           0           None        -3               0.9968  MODERATE   \n",
       "\n",
       "        consequence  \n",
       "0  missense_variant  "
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_df = pd.concat([dataset_cp,features_df],axis=1) \n",
    "test_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "06056285",
   "metadata": {},
   "source": [
    "### Purposefully defined test dataset that will serve for EDA\n",
    "(LoF score was not available to retrieve, so it was excluded from the start)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "bd6c076b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cadd_phred</th>\n",
       "      <th>cadd_raw</th>\n",
       "      <th>revel</th>\n",
       "      <th>sift_score</th>\n",
       "      <th>polyphen_score</th>\n",
       "      <th>blosum62</th>\n",
       "      <th>alphamissense_score</th>\n",
       "      <th>impact</th>\n",
       "      <th>consequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>29.2</td>\n",
       "      <td>5.203168</td>\n",
       "      <td>0.927</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>-3</td>\n",
       "      <td>0.9968</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>missense_variant</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   cadd_phred  cadd_raw  revel  sift_score polyphen_score  blosum62  \\\n",
       "0        29.2  5.203168  0.927           0           None        -3   \n",
       "\n",
       "   alphamissense_score    impact       consequence  \n",
       "0               0.9968  MODERATE  missense_variant  "
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "features_df = test_df.iloc[:,8:]\n",
    "features_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b5353b18",
   "metadata": {},
   "source": [
    "### And target:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "e56c12ed",
   "metadata": {},
   "outputs": [],
   "source": [
    "target = test_df['ClinSigSimple']\n",
    "features_df['Target'] = target"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ab88c98c",
   "metadata": {},
   "source": [
    "features + target:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "d4f0f349",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cadd_phred</th>\n",
       "      <th>cadd_raw</th>\n",
       "      <th>revel</th>\n",
       "      <th>sift_score</th>\n",
       "      <th>polyphen_score</th>\n",
       "      <th>blosum62</th>\n",
       "      <th>alphamissense_score</th>\n",
       "      <th>impact</th>\n",
       "      <th>consequence</th>\n",
       "      <th>Target</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>29.2</td>\n",
       "      <td>5.203168</td>\n",
       "      <td>0.927</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>-3</td>\n",
       "      <td>0.9968</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   cadd_phred  cadd_raw  revel  sift_score polyphen_score  blosum62  \\\n",
       "0        29.2  5.203168  0.927           0           None        -3   \n",
       "\n",
       "   alphamissense_score    impact       consequence  Target  \n",
       "0               0.9968  MODERATE  missense_variant       1  "
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "features_df.head(35)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8aa27fa3",
   "metadata": {},
   "source": [
    "Next step:\n",
    "Complete retrieval of batch annotations from the REST API"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4b006f5b",
   "metadata": {},
   "source": [
    "U prilogu se nalazi kompletan dataset spreman za EDA (POST request ima limit od 200)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e481ae72",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "ac9d07b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "def preuzmi_anotacije(batch_list): \n",
    "    server = \"https://rest.ensembl.org\" \n",
    "    ext = \"/vep/homo_sapiens/region\"\n",
    "\n",
    "    headers = { \n",
    "        \"Content-Type\": \"application/json\",\n",
    "        \"Accept\": \"application/json\"\n",
    "    } \n",
    "\n",
    "    variant = batch_list  \n",
    "\n",
    "    data = {\n",
    "        \"variants\": variant\n",
    "    } \n",
    "    param ={ \n",
    "        'canonical':1, \n",
    "        'CADD':1, \n",
    "        'AlphaMissense':1,\n",
    "        'Blosum62':1,\n",
    "        'LoF':1,\n",
    "        'REVEL':1\n",
    "    }\n",
    "    try:\n",
    "        r = requests.post(server + ext, headers=headers, json=data,params=param)\n",
    "        r.raise_for_status()\n",
    "        return r.json()\n",
    "    except requests.exceptions.HTTPError as err:\n",
    "        print(f\"HTTP Error: {err}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "607512ba",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HTTP Error: 503 Server Error: Service Unavailable for url: https://rest.ensembl.org/vep/homo_sapiens/region?canonical=1&CADD=1&AlphaMissense=1&Blosum62=1&LoF=1&REVEL=1\n"
     ]
    },
    {
     "ename": "TypeError",
     "evalue": "'NoneType' object is not iterable",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mTypeError\u001b[39m                                 Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[51]\u001b[39m\u001b[32m, line 6\u001b[39m\n\u001b[32m      4\u001b[39m batch = variantsParams[i:i + batch_size] \u001b[38;5;66;03m#  150, 200 limit\u001b[39;00m\n\u001b[32m      5\u001b[39m batch_result = preuzmi_anotacije(batch)\n\u001b[32m----> \u001b[39m\u001b[32m6\u001b[39m \u001b[43mstorage\u001b[49m\u001b[43m.\u001b[49m\u001b[43mextend\u001b[49m\u001b[43m(\u001b[49m\u001b[43mbatch_result\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[31mTypeError\u001b[39m: 'NoneType' object is not iterable"
     ]
    }
   ],
   "source": [
    "storage = []\n",
    "batch_size = 150\n",
    "for i in range(0,len(variantsParams),batch_size):\n",
    "    batch = variantsParams[i:i + batch_size] #  150, 200 limit\n",
    "    batch_result = preuzmi_anotacije(batch)\n",
    "    storage.extend(batch_result)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e8961318",
   "metadata": {},
   "outputs": [],
   "source": [
    "features_kompletno = []\n",
    "for i,variant in enumerate(storage):\n",
    "    if not isinstance(variant,dict):\n",
    "        continue\n",
    "    if 'transcript_consequences' not in variant:\n",
    "        continue\n",
    "    transcripti = variant['transcript_consequences']\n",
    "    selected = select_transcript(transcripti)\n",
    "\n",
    "    features = {\n",
    "        'variant_id': i+1,\n",
    "        'cadd_phred': selected.get('cadd_phred'),\n",
    "        'cadd_raw': selected.get('cadd_raw'),\n",
    "        'revel': selected.get('revel'),\n",
    "        'sift_score': selected.get('sift_score'),\n",
    "        'blosum62': selected.get('blosum62'),\n",
    "        'alphamissense_score': selected.get('alphamissense', {}).get('am_pathogenicity'),\n",
    "        'impact': selected.get('impact'),\n",
    "        'consequence': selected.get('consequence_terms', [None])[0]\n",
    "    }\n",
    "    \n",
    "    features_kompletno.append(features)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b82f5e62",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2647\n"
     ]
    }
   ],
   "source": [
    "print(features_kompletno[-1]['variant_id'])\n",
    "# 2647 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f3134554",
   "metadata": {},
   "outputs": [],
   "source": [
    "svi_redovi = []\n",
    "for i in features_kompletno:\n",
    "    ime_kolone = i.keys()\n",
    "    redovi = i.values()\n",
    "    svi_redovi.append(list(redovi))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bf5c7508",
   "metadata": {},
   "outputs": [],
   "source": [
    "to_append = pd.DataFrame(data=svi_redovi,columns=ime_kolone)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7c503aef",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2646, 9)"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# dataset with annotations\n",
    "to_append.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4a953c7d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start</th>\n",
       "      <th>ReferenceAlleleVCF</th>\n",
       "      <th>AlternateAlleleVCF</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>ClinSigSimple</th>\n",
       "      <th>vcf_format</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17</td>\n",
       "      <td>7674221</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1</td>\n",
       "      <td>17 7674221 . G A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>17</td>\n",
       "      <td>7674191</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1</td>\n",
       "      <td>17 7674191 . C T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>17</td>\n",
       "      <td>7674230</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1</td>\n",
       "      <td>17 7674230 . C A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>17</td>\n",
       "      <td>7674208</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1</td>\n",
       "      <td>17 7674208 . A G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>17</td>\n",
       "      <td>7676154</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1</td>\n",
       "      <td>17 7676154 . G C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2658</th>\n",
       "      <td>17</td>\n",
       "      <td>7676070</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>TP53</td>\n",
       "      <td>0</td>\n",
       "      <td>17 7676070 . T G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2659</th>\n",
       "      <td>17</td>\n",
       "      <td>7673762</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>TP53</td>\n",
       "      <td>0</td>\n",
       "      <td>17 7673762 . T C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2660</th>\n",
       "      <td>17</td>\n",
       "      <td>7669609</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>TP53</td>\n",
       "      <td>0</td>\n",
       "      <td>17 7669609 . T G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2661</th>\n",
       "      <td>17</td>\n",
       "      <td>7675221</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1</td>\n",
       "      <td>17 7675221 . T G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2662</th>\n",
       "      <td>17</td>\n",
       "      <td>7677724</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>TP53</td>\n",
       "      <td>0</td>\n",
       "      <td>17 7677724 . A G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2647 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chromosome    Start ReferenceAlleleVCF AlternateAlleleVCF GeneSymbol  \\\n",
       "0            17  7674221                  G                  A       TP53   \n",
       "1            17  7674191                  C                  T       TP53   \n",
       "2            17  7674230                  C                  A       TP53   \n",
       "3            17  7674208                  A                  G       TP53   \n",
       "4            17  7676154                  G                  C       TP53   \n",
       "...         ...      ...                ...                ...        ...   \n",
       "2658         17  7676070                  T                  G       TP53   \n",
       "2659         17  7673762                  T                  C       TP53   \n",
       "2660         17  7669609                  T                  G       TP53   \n",
       "2661         17  7675221                  T                  G       TP53   \n",
       "2662         17  7677724                  A                  G       TP53   \n",
       "\n",
       "      ClinSigSimple        vcf_format  \n",
       "0                 1  17 7674221 . G A  \n",
       "1                 1  17 7674191 . C T  \n",
       "2                 1  17 7674230 . C A  \n",
       "3                 1  17 7674208 . A G  \n",
       "4                 1  17 7676154 . G C  \n",
       "...             ...               ...  \n",
       "2658              0  17 7676070 . T G  \n",
       "2659              0  17 7673762 . T C  \n",
       "2660              0  17 7669609 . T G  \n",
       "2661              1  17 7675221 . T G  \n",
       "2662              0  17 7677724 . A G  \n",
       "\n",
       "[2647 rows x 7 columns]"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# pocetni dataset\n",
    "dataset.shape\n",
    "dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "55034b36",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "WindowsPath('c:/Users/Admin/ML/tp53_variant_classification/notebooks')"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Path.cwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3c79e44e",
   "metadata": {},
   "outputs": [],
   "source": [
    "complete_dataset = pd.concat([dataset,to_append],axis=1)\n",
    "complete_dataset.to_csv('c:/Users/Admin/ML/tp53_variant_classification/data/processed/dataset_raw.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "80c43e69",
   "metadata": {},
   "outputs": [],
   "source": [
    "complete_dataset.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bee8331f",
   "metadata": {},
   "source": [
    "---"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv1",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
